
# 2022-1109
<details>
<summary><b><font size="+2">Table of Contents</font></b></summary>
<!-- MarkdownTOC -->

1. [Reading and taking notes on the papers sent by Alison](#reading-and-taking-notes-on-the-papers-sent-by-alison)
1. [Reading through Alison's qualifying exam \(QE\) proposal](#reading-through-alisons-qualifying-exam-qe-proposal)
    1. [QE proposal *Introduction*](#qe-proposal-introduction)
    1. [QE proposal *Aims*](#qe-proposal-aims)
    1. [QE proposal *Background and Significance*](#qe-proposal-background-and-significance)
    1. [QE proposal *Approach*](#qe-proposal-approach)
        1. [Aim 1: Determine Function of Nuclear Exosome Cofactor *TRF4* in Quiescence Entry](#aim-1-determine-function-of-nuclear-exosome-cofactor-trf4-in-quiescence-entry)
            1. [Aim 1.1: Which transcripts misregulated by TRF4 are most essential to Q entry?](#aim-11-which-transcripts-misregulated-by-trf4-are-most-essential-to-q-entry)
                1. [Aim 1.1: Expected Outcomes](#aim-11-expected-outcomes)
                1. [Aim 1.1: Pitfalls and Alternative Approaches](#aim-11-pitfalls-and-alternative-approaches)
            1. [Aim 1.2: What are the consequences of undegraded RNAs in Q entry?](#aim-12-what-are-the-consequences-of-undegraded-rnas-in-q-entry)
                1. [Aim 1.2: Expected Outcomes](#aim-12-expected-outcomes)
                1. [Aim 1.2: Pitfalls and Alternative Approaches](#aim-12-pitfalls-and-alternative-approaches)
            1. [Aim 1.3: What effect does *trf4Δ* have on transcription in Q entry?](#aim-13-what-effect-does-trf4%CE%94-have-on-transcription-in-q-entry)
                1. [Aim 1.3: Expected Outcomes](#aim-13-expected-outcomes)
                1. [Aim 1.3: Pitfalls and Alternative Approaches](#aim-13-pitfalls-and-alternative-approaches)
        1. [Aim 2: Investigate Functional Antisense Transcription in Quiescence](#aim-2-investigate-functional-antisense-transcription-in-quiescence)
            1. [Aim 2.1: Nascent RNA sequencing and Annotation of Quiescent Transcriptome](#aim-21-nascent-rna-sequencing-and-annotation-of-quiescent-transcriptome)
                1. [Aim 2.1: Expected Outcomes](#aim-21-expected-outcomes)
                1. [Aim 2.1: Pitfalls and Alternative approaches](#aim-21-pitfalls-and-alternative-approaches)
            1. [Aim 2.2: Transcriptional interference in Quiescence](#aim-22-transcriptional-interference-in-quiescence)
                1. [Aim 2.2: Expected Outcomes](#aim-22-expected-outcomes)
                1. [Aim 2.2: Pitfalls and Alternative approaches](#aim-22-pitfalls-and-alternative-approaches)
    1. [QE proposal *Future Directions*](#qe-proposal-future-directions)
    1. [QE proposal *Rigor and Reproducibility*](#qe-proposal-rigor-and-reproducibility)
    1. [QE proposal *Innovation*](#qe-proposal-innovation)
1. [Reading through Alison's Research Summary, Spring 2022](#reading-through-alisons-research-summary-spring-2022)
    1. [Role of Nuclear Exosome co-factor Trf4 in Quiescence Entry](#role-of-nuclear-exosome-co-factor-trf4-in-quiescence-entry)
    1. [Nascent Transcription in Quiescence Project](#nascent-transcription-in-quiescence-project)

<!-- /MarkdownTOC -->
</details>
<br />

<a id="reading-and-taking-notes-on-the-papers-sent-by-alison"></a>
## Reading and taking notes on the papers sent by Alison
`#TODO` Copy in email here

<a id="reading-through-alisons-qualifying-exam-qe-proposal"></a>
## Reading through Alison's qualifying exam (QE) proposal
<a id="qe-proposal-introduction"></a>
### QE proposal *Introduction*
¶2  
A conserved feature of the eukaryotic genomes from yeast to humans is pervasive transcription. In *S. cerevisiae* <mark>~75% of the genome is comprised of annotated genes</mark>, and <mark>at least 85% is transcribed</mark>.3 This results in <mark>noncoding transcription which is either in close proximity to or **overlaps** annotated genes</mark>, much of it using the antisense strand. However, many of the transcripts produced are <mark>rapidly degraded</mark>, therefore termed <mark>“cryptic”</mark>. These transcripts have largely been annotated and classified based on what factor was knocked out in order to stabilize them, but *comprehensive study of their biological function remains a challenge*. The extent to which cryptic transcription is functional remains a matter of debate. Antisense transcription has been found to be a weak regulator of protein levels – regulating ~25% of 188 genes tested.4

```txt
Cryptic transcription
    - 85% of genes are transcribed in cycling yeast cells, resulting in substantial amounts
      of noncoding transcription; however, much of this noncoding transcription is quickly
      degraded, before it can be detected with, e.g., RNA-seq or similar NGS methods
    - It is this undetected transcription (which is degraded on a much shorter time scale
      than transcription that can be detected by RNA-seq) that is termed cryptic
      transcription
```

¶3  
... My preliminary analysis shows that Q yeast show differentially expressed antisense transcription at ~600 loci compared to G1 arrested cells. As a result, quiescence is an ideal state to enrich for functional ncRNAs and noncoding transcription.

¶4  
Many noncoding RNAs are rapidly degraded. Much of this degradation is controlled by the nuclear exosome. The nuclear exosome has been linked to genetic disease and cancer.9 Additionally, RNA stability is highly regulated through the cell cycle10 and contribute to a fibroblast model of quiescence11. However, how RNA stability is regulated in Q entry has not been comprehensively studied. ...

<a id="qe-proposal-aims"></a>
### QE proposal *Aims*
¶1  
**Aim 1: Determine Function of Nuclear Exosome Cofactor TRF4 in Quiescence Entry**  
`#NOTE Per Alison, this aim is dead now; see the email in which she sent this write-up`  
`#TODO Paste that info in here`

¶2  
**Aim 2: Investigate Functional Antisense Transcription in Quiescence**  
In Aim 2, I will use nascent RNA sequencing as a discovery mechanism for functional antisense transcription in quiescence. In Aim 2.1, I will focus on the analysis of nascent RNA sequencing in Q and G1 arrested cells. The goal of which is to select several candidates to test for functionality in Aim 2.2. I will test 6 to 10 candidates for functionality using CRISPRi and RNA depletion via hammerhead ribozyme to target transcription and RNA level separately. This will determine if antisense transcription functions to repress sense transcription in Q.

<a id="qe-proposal-background-and-significance"></a>
### QE proposal *Background and Significance*
¶1  
...

¶2  
...

¶3-5: *Conserved Nuclear Surveillance Factor TRF4 is Essential to Q Entry*  
...  
```txt
The following is a note from me:
```
She refers to *"the fibroblast model of quiescence"* a few times: what is this?

From the abstract of the paper on it, [Johnson, Robinson, and Coller, *BMC Genomics* 2017](https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-017-3521-0):  
*Background*  
Quiescence, reversible exit from the cell division cycle, is characterized by large-scale changes in steady-state gene expression, yet mechanisms controlling these changes are in need of further elucidation. In order to characterize the effects of post-transcriptional control on the quiescent transcriptome in human fibroblasts, we determined mRNA decay rates for over 10,000 genes using a transcription shut-off time-course.

*Results*  
We found that ~500 of the genes monitored exhibited significant changes in decay rate upon quiescence induction. Genes involved in RNA processing and ribosome biogenesis were destabilized with quiescence, while genes involved in the developmental process were stabilized with quiescence. Moreover, extracellular matrix genes demonstrated an upregulation of gene expression that corresponded with a stabilization of these transcripts. Additionally, targets of a quiescence-associated microRNA (miR-29) were significantly enriched in the fraction of transcripts that were stabilized during quiescence.

*Conclusion*  
Coordinated stability changes in clusters of genes with important functions in fibroblast quiescence maintenance are highly correlated with quiescence gene expression patterns. Analysis of miR-29 target decay rates suggests that microRNA-induced changes in RNA stability are important contributors to the quiescence gene expression program in fibroblasts. The identification of multiple stability-related gene clusters suggests that other posttranscriptional regulators of transcript stability may contribute to the coordination of quiescence gene expression. Such regulators may ultimately prove to be valuable targets for therapeutics that target proliferative cells, for instance, in cancer or fibrosis.
```txt
Back to Alison:
```
¶6-X: *Regulatory Noncoding Transcription in Quiescence*  
Pervasive transcription is a conserved feature of the eukaryotic transcriptome from yeast to humans. In S. cerevisiae ~75% of the genome is comprised of annotated genes, and at least 85% is transcribed.3
```txt
What is reference 3? Seems important...

The following is text from me:
```
Reference 3 is [David et al. (Steinmetz), *PNAS* 2006](https://www.pnas.org/doi/10.1073/pnas.0601091103)

From its abstract:  
There is abundant transcription from eukaryotic genomes unaccounted for by protein coding genes. A high-resolution genome-wide survey of transcription in a well annotated genome will help relate transcriptional complexity to function. By quantifying RNA expression on both strands of the complete genome of *Saccharomyces cerevisiae* using <mark>a high-density oligonucleotide tiling array</mark>, this study identifies the boundary, structure, and level of coding and noncoding transcripts. <mark>A total of 85% of the genome is expressed in rich media</mark>. Apart from expected transcripts, we found <mark>operon-like transcripts, transcripts from neighboring genes not separated by intergenic regions, and genes with complex transcriptional architecture where different parts of the same gene are expressed at different levels</mark>. We mapped the positions of 3′ and 5′ UTRs of coding genes and <mark>identified hundreds of RNA transcripts distinct from annotated genes</mark>. These nonannotated transcripts, on average, have lower sequence conservation and lower rates of deletion phenotype than protein coding genes. <mark>Many other transcripts overlap known genes in antisense orientation, and for these pairs global correlations were discovered: UTR lengths correlated with gene function, localization, and requirements for regulation; **antisense transcripts overlapped 3’ UTRs more than 5’ UTRs**; **UTRs with overlapping antisense tended to be longer**; and **the presence of antisense associated with gene function**</mark>. These findings may suggest a regulatory role of antisense transcription in *S. cerevisiae*. Moreover, the data show that even this well studied genome has transcriptional complexity far beyond current annotation.
```txt
Back to Alison:
```
... Annotation of \[cryptic transcription\] has been done largely by removing RNA surveillance factors/chromatin modifiers and annotating the resulting stabilized transcripts.

These include XUTs33 (Xrn1 unstable transcripts), CUTs20,34,35 (Cryptic Unstable Transcripts, Rrp6 and/or Trf4), NUTs36 (Nrd1 Unterminated Transcripts), SUTs37 (Stable Unannotated Transcripts), and SRATs38 (Set2 repressed antisense transcripts).

While annotation has been widespread, most of these units of transcription have not been studied functionally.
```txt
If the annotation is widespread, are there .gtf files (or the like) that we can get hold of?
```

The extent to which cryptic transcription is functional remains a matter of debate. Antisense transcription has been found to be a weak regulator of protein levels – regulating ~25% of 188 tested transcripts, though this may be the result of condition specific regulation.4

¶7  
<mark>***The loss of RNAi in the evolution of S. cerevisiae has resulted in an expansion of antisense lncRNA.39***</mark>

However, this mechanism of regulation does persist in species with RNAi.40 Antisense transcription has been repeatedly been found to be a regulator of cell fate and metabolic state in budding yeast. This regulation acts in *cis*, as functional transcription rather than as functional RNA product. For example, IME4 is regulated by an antisense transcription which determines meiotic entry.41 At the Gal locus, antisense transcription contributes to inhibition of *GAL10* and *GAL1* in the absence of galactose.42 Additionally, a number of Q expressed mRNAs are repressed by antisense RNAs during logarithmic growth, and one locus, *PET10* was shown to function in *cis*.43 Further, some antisense transcription has been shown to be conserved across 5 yeast species, separated by 150 million years of evolution.8

¶8  
While there are examples of functional antisense loci, to date there have not been any clear functional antisense RNAs found in *S. cerevisiae*. Instead, antisense ncRNA in budding yeast has been found to function on the level of transcription, by transcription interreference. Transcriptional interference is transcription that functions to block other transcription either by promotor competition, occlusion, or collision.44 *PHO84* was originally thought to be an example of a functional transcript, as the stabilization of its antisense RNA in *rrp6Δ* led to a decrease in sense transcription.45 However, further study showed that *rrp6Δ* has PolII termination defects, and thus *PHO84* sense transcription was found to be regulated by antisense transcription activity and not by functional antisense RNA product.46 While some antisense transcription has been studied, it is likely there are yet undiscovered roles for this pervasive transcription. Much of \[antisense\] transcription may only be regulatory in some conditions; at *GAL1* and *GAL10*, antisense RNA is spurious in the presence of galactose and functional only in its absence.42 
```txt
The above paragraph describes the distinction between antisense regulation ("functional antisense loci") via
    - (a) antisense transcription activity
    - (b) functional antisense RNA product

...and there are no "clear" examples of the latter in S. cerevisiae

Also, "budding yeast" is a synonym for S. cerevisiae; see "Reproduction" in en.wikipedia.org/wiki/Yeast
```

¶9  
In Aim 1, I will <mark>determine how steady state RNA levels change though Q entry</mark>, and the essential role TRF4 plays in this process. In Aim 2, I will <mark>annotate unique noncoding and antisense transcripts in Quiescence</mark> and test candidates for functionality.

<a id="qe-proposal-approach"></a>
### QE proposal *Approach*
<a id="aim-1-determine-function-of-nuclear-exosome-cofactor-trf4-in-quiescence-entry"></a>
#### Aim 1: Determine Function of Nuclear Exosome Cofactor *TRF4* in Quiescence Entry
...

<a id="aim-11-which-transcripts-misregulated-by-trf4-are-most-essential-to-q-entry"></a>
##### Aim 1.1: Which transcripts misregulated by TRF4 are most essential to Q entry?
...

<a id="aim-11-expected-outcomes"></a>
###### Aim 1.1: Expected Outcomes
...

<a id="aim-11-pitfalls-and-alternative-approaches"></a>
###### Aim 1.1: Pitfalls and Alternative Approaches
...

<a id="aim-12-what-are-the-consequences-of-undegraded-rnas-in-q-entry"></a>
##### Aim 1.2: What are the consequences of undegraded RNAs in Q entry?
...

<a id="aim-12-expected-outcomes"></a>
###### Aim 1.2: Expected Outcomes
...

<a id="aim-12-pitfalls-and-alternative-approaches"></a>
###### Aim 1.2: Pitfalls and Alternative Approaches
...

<a id="aim-13-what-effect-does-trf4%CE%94-have-on-transcription-in-q-entry"></a>
##### Aim 1.3: What effect does *trf4Δ* have on transcription in Q entry?
...

<a id="aim-13-expected-outcomes"></a>
###### Aim 1.3: Expected Outcomes
...

<a id="aim-13-pitfalls-and-alternative-approaches"></a>
###### Aim 1.3: Pitfalls and Alternative Approaches
...

<a id="aim-2-investigate-functional-antisense-transcription-in-quiescence"></a>
#### Aim 2: Investigate Functional Antisense Transcription in Quiescence
Functional antisense RNAs have been a challenge to find and characterize. Quiescence provides a unique cell state with the possibility of enriching for such functional transcripts. This is because antisense transcripts have repeatedly been linked to regulation of cell fate, and <mark>Q has widespread induction of antisense RNA</mark>. Further, antisense transcription regulation is a promising candidate for regulation of quiescence due to ties to metabolic and cell state, as well as evolutionary conservation. <mark>We have found in Q an increase in noncoding transcription in general, and antisense transcription specifically. I hypothesize that antisense transcription regulates key loci via transcriptional interference to help maintain quiescent genome off state</mark>.

<a id="aim-21-nascent-rna-sequencing-and-annotation-of-quiescent-transcriptome"></a>
##### Aim 2.1: Nascent RNA sequencing and Annotation of Quiescent Transcriptome
¶1  
In order to understand transcription in quiescence, I performed nascent RNA sequencing on quiescent and G1 arrested yeast. Q cells in water or G1 arrested cells in media were given 4tU for 6 minutes before being snap frozen. Labeled K. lactis cells were added at the time of RNA extraction as a spike-in. 40 ng of RNA was then conjugated to biotin, which preferentially binds with the sulfur on 4tU. Nascent RNA is then enriched for by bead pull down, which pulls down biotinylated nascent RNA. Both input (steady state) and pulled down (nascent) RNA were then library prepped and sequenced using the strand specific kit.

¶2  
Preliminary analysis shows ~600 of a total 6600 yeast genes have differentially expressed antisense transcription compared to G1. Antisense transcription represents 2-3% of total aligned nascent reads, about double that of G1. Antisense and sense transcription are not mutually exclusive, and many loci are transcribed in both directions.

¶3  
<mark>Genome annotation in yeast has largely been performed in log cells. Therefore, units of transcription unique to Q have not been annotated</mark>. I will annotate transcription *de novo* using Stringtie. I will use stringent gating to make sure transcripts are significantly enriched above expression in G1.
```txt
#QUESTION What does this ("stringent gating") mean?
```
Additionally, I will assess antisense transcription. I will quantify the extent of antisense transcription, and to what extent it anti-correlates to sense transcription.
```txt
#QUESTION Quantify antisense transcription how?
```
Once annotation is complete, I will classify transcripts. For antisense transcripts I will classify by length and stability. For length there will be 3 categories: 3’ end only, full gene and into sense promotor. For stability, I will determine if transcripts are rapidly degraded or not; if a transcript is not above a certain threshold in input, it will be classified as rapidly degraded.
```txt
Antisense transcript classification schema:
    I. By length
        a. 3’ end only
        b. full gene
        c. into sense promotor
    II. By stability: "if a transcript is not above a certain threshold in input, it will be
        classified as rapidly degraded"

#QUESTION For II, what is the input? How will it be measured, assessed, and thresholded?
```

¶4  
Based on this analysis I will select candidate antisense transcription units to test for functionality in Aim 2.2. I will use several criteria to assess what transcripts are of most interest. These are high antisense transcription, uniqueness to Q, and potential biological effect on Q if sense strand is transcribed. I will further favor candidates which either extend across the full length of sense gene or where shorter 3’ transcript looks like it could be inducing early termination of sense transcription. I will include some transcripts which are rapidly degraded, and some which are not; since I hypothesize that transcription rather than the RNA itself is functional, a transcript need not be stable to be of interest.
```txt
I. Schema for assessing "the interest" of antisense transcripts:
    a. high antisense transcription
    b. uniqueness to Q
    c. potential biological effect on Q if sense strand is transcribed

II. Home in on candidates from the above that...
    a. extend across the full length of sense gene
    b. appears to be "shorter 3’ transcript [that] looks like it could be inducing early
       termination of sense transcription"

#QUESTION How is antisense transcription evaluated to be "high" in Ia?
#QUESTION How to assess Ic?
#QUESTION How can you tell IIb just by looking at IGV tracks? You see the end of sense
          transcription in the 3' UTR together with the antisense transcription specific
          to/overlapping the 3' UTR?
```

¶5  
Since antisense RNAs are often transcribed alongside sense and often result from tandem genes, where antisense could easily represent leaky transcription. This phenomenon is illustrated in figure 3; these three loci contribute to regulation of metabolic state (*HIS1* and *MIG2*) or function in mating (*SAG1*). <mark>While antisense RNA may be spurious rather functional, the fact that many genes have functions in metabolism and cell state is a promising indication that they may function to repress sense transcription.</mark> I will further enrich for functionality by selecting antisense RNAs with unique pre-initiation complexes. I will do CUT&RUN of core transcription factor TFIIB. I will then pick candidates enriched for those with TFIIB sites distinct from sense transcripts, as this implies a uniquely regulated locus.
```txt
Here, Figure 3 is a mistake and means Figure 4, I think

#QUESTION The term "tandem genes" returns multiple definitions when I Google it: What does
          Alison mean when she writes it here?
#QUESTION How to determine "antisense RNAs with unique pre-initiation complexes"? From the
          literature? Or the subsequent TFIIB thing? Both?
```
<a id="aim-21-expected-outcomes"></a>
###### Aim 2.1: Expected Outcomes
I expect to identify 6-10 strong candidate antisense RNAs based on the approach outlined above.

<a id="aim-21-pitfalls-and-alternative-approaches"></a>
###### Aim 2.1: Pitfalls and Alternative approaches
4tU sequencing data is high quality, and there is strong correlation between biological replicates and to similar data generated by former lab members. The biggest challenge is selecting candidate transcripts most likely to have regulatory functions. I have already identified some exciting loci. However, if further analysis suggests this phenomenon is not functional, I will shift focus to understand how the chromatin environment in Q results in a higher fraction of pervasive transcription (Future Directions).
```txt
#QUESTION Do we have data on the quality of the 4tU datasets? My understanding is that these
          have not been done yet?
#QUESTION Are there data on the evaluation (e.g., correlation) of Alison's data with similar
          data from other lab members?
```

<a id="aim-22-transcriptional-interference-in-quiescence"></a>
##### Aim 2.2: Transcriptional interference in Quiescence
...

<a id="aim-22-expected-outcomes"></a>
###### Aim 2.2: Expected Outcomes
...

<a id="aim-22-pitfalls-and-alternative-approaches"></a>
###### Aim 2.2: Pitfalls and Alternative approaches
...

<a id="qe-proposal-future-directions"></a>
### QE proposal *Future Directions*
In the future, I would like to see when functional antisense RNAs come on in Q entry; this may be observable in RNA seq data generated in Aim 1 or via strand specific RT-qPCR at specific loci. Additionally, to further understand how the Q chromatin environment contributes to noncoding transcription, I will compare antisense transcripts to features already ChIPed by our lab and others to see how the chromatin environment in quiescence specifically promotes widespread antisense transcription. This along with additional bioinformatic analysis such as motif analysis, as well as analysis of gene orientation has the potential to reveal organizing principles of antisense transcription in Q.

<a id="qe-proposal-rigor-and-reproducibility"></a>
### QE proposal *Rigor and Reproducibility*
...

<a id="qe-proposal-innovation"></a>
### QE proposal *Innovation*
...
<br />
<br />

<a id="reading-through-alisons-research-summary-spring-2022"></a>
## Reading through Alison's Research Summary, Spring 2022
<a id="role-of-nuclear-exosome-co-factor-trf4-in-quiescence-entry"></a>
### Role of Nuclear Exosome co-factor Trf4 in Quiescence Entry
¶1  
I sequenced steady state RNA from *trf4∆* and parental strains through quiescence entry. Additionally, I grew *rrp6∆* strains in parallel. Since *RRP6* functions in the same pathway as *TRF4*, but with much less dramatic effects on Q entry, this provided a key control to understand which RNAs are most relevant to Q entry. Overall, the transcriptome between *trf4∆* and *rrp6∆* are much more similar to each other and to the parental control than I anticipated. I hypothesized that TRF4 functioned mainly to degrade growth mRNAs in Q entry, which was consistent with previous work done in rapid nutrient downshift. While hundreds of mRNAs are preferentially stabilized, those defects are less dramatic in *trf4∆* than *rrp6∆*, and don’t seem to target pathways that would affect Q. Instead, I have found a handful mRNAs which regulated phosphate and nitrogen metabolism are down in *trf4∆* two hours before the diauxic shift (DS). Coordinating the starvation response of multiple nutrients is necessary for entry into quiescence, so this is potentially explanatory of the phenotype we observed. By 24 hours after the DS, *trf4∆* overexpresses a number of glucose transporters despite having exhausted glucose 24 hours prior. My current working hypothesis is that failure to coordinate nutrient starvation response early in Q entry results in a cellular state incompatible with quiescence for *trf4∆*.

¶2  
However, this opens a new line of questioning: how is the deletion of an RNA decay protein resulting in less transcript not more? My immediate next step is to use nascent RNAseq to understand if this is change in RNA abundance is being driven by more decay or less transcription in the absence of Trf4.

<a id="nascent-transcription-in-quiescence-project"></a>
### Nascent Transcription in Quiescence Project
¶1  
Annotation of unique quiescent transcripts is complete, and I have identified 5447 transcripts. When compared to an annotation file with 14623 annotated yeast transcripts, 1699 transcripts had no overlap on the same strand with anything previously annotated. This annotation was therefore ~31% “totally unique”. Trinity, the program which performed the initial annotation, is very sensitive, so some of these 5000 transcripts are very low level, and some are fragmented. Using DeSeq2, with cut offs of log2 fold change >2 and adjusted p-value < .05, 61% of newly annotated transcripts are overexpressed in Q compared to G1 arrested cells.

¶2  
My focus since the completion of this annotation has been seeking to define classes of transcripts as well as to determine which are of highest interest for further study. My annotations include 1275 intergenic transcripts, 673 extended transcripts (transcripts which are 200 bp extended in either the 3’ or 5’ direction compared to their SGD annotation), and approximately 1700 antisense transcripts. As I was breaking the main annotations into general classes, I noticed three main classes of antisense transcripts which I have been investigating in further detail.

¶3  
Those three classes of antisense transcripts are defined by their promoters (see Figure 1). Class 1 transcripts originate from tandem intergenic regions; the promoter for the adjacent sense transcript is near that of the antisense. Class 2 antisense transcripts arise from convergent intergenic regions where less efficient transcription termination results in the 3’ end of the sense transcript becoming antisense to its neighbor. Finally, Class 3 antisense transcripts arise from the general location of transcript termination for the sense transcript. These originate from genes with large 3’ intergenic regions. By my current estimates, there are 533 Class 1 transcripts, 217 Class 2 intergenic regions, and 286 Class 3 transcripts.

¶4  
I have spent a great deal of time computationally analyzing these three classes of transcripts. Previous lab members have profiled the chromatin state in Q including H3-ChIP, MNase seq, TFIIB ChIP and ChIP of every chromatin remodeler in Q. Using this data as well GO terms and expression data, I failed to find a smoking gun which suggests which transcripts are most likely to be functional. To identify antisense transcripts that have effects on mRNAs, I will next deplete Nrd1 in Q cells to increase antisense transcription, and using Nascent RNAseq determine which mRNAs go down when their corresponding antisense transcript goes up.
