---
title: "work_assess_R64-1-1_gff3.Rmd"
output: html_notebookauthor: "KA"
email: "kalavatt@fredhutch.org"
output:
    html_notebook:
        toc: yes
        toc_float: true
---
<br />

Want to see how this file from SGD differs from the Ensembl 108 file we have been using
`#TODO` Include file names, more info

## Get situated
### Code
<details>
<summary><i>Code: Get situated</i></summary>
```{r}
#!/usr/bin/env Rscript

library(GenomicRanges)
library(IRanges)
library(readxl)
library(rtracklayer)
library(tidyverse)

options(scipen = 999)
options(ggrepel.max.overlaps = Inf)

if(stringr::str_detect(getwd(), "kalavattam")) {
    p_local <- "/Users/kalavattam/Dropbox/FHCC"
} else {
    p_local <- "/Users/kalavatt/projects-etc"
}
p_wd <- "2022_transcriptome-construction/results/2023-0215"

setwd(paste(p_local, p_wd, sep = "/"))
getwd()

rm(p_local, p_wd)
```
</details>
<br />
<br />

## Load `gtf`, `gff3` files
### Code
<details>
<summary><i>Code: Load "R64" (`saccharomyces_cerevisiae_R64-1-1_20110208.gff`)</i></summary>
```{r Load "R64", results='hide', message=FALSE, warning=FALSE}
p_SGD <- "./infiles_gtf-gff3/comprehensive/S288C_reference_genome_R64-1-1_20110203"
f_SGD <- "saccharomyces_cerevisiae_R64-1-1_20110208.gff"
gff3_SGD <- paste(p_SGD, f_SGD, sep = "/") %>%
    rtracklayer::import() %>%
    tibble::as_tibble() %>%
    dplyr::arrange(seqnames, start) %>%
    dplyr::mutate(seqnames = gsub("chr", "", seqnames))

rm(p_SGD, f_SGD)

p_Ens <- "./infiles_gtf-gff3/already"
f_Ens <- "Saccharomyces_cerevisiae.R64-1-1.108.gff3"
gff3_Ens <- paste(p_Ens, f_Ens, sep = "/") %>%
    rtracklayer::import() %>%
    tibble::as_tibble() %>%
    dplyr::arrange(seqnames, start)

rm(p_Ens, f_Ens)

p_Ens <- "./infiles_gtf-gff3/already"
f_Ens <- "Saccharomyces_cerevisiae.R64-1-1.108.gtf"
gtf_Ens <- paste(p_Ens, f_Ens, sep = "/") %>%
    rtracklayer::import() %>%
    tibble::as_tibble() %>%
    dplyr::arrange(seqnames, start)

rm(p_Ens, f_Ens)
```
</details>
<br />
<br />

## Perform comparisons
### Are the numbers of sn/snoRNA features per `gtf`/`gff3` the same?
#### Code
<details>
<summary><i>Code: Are the numbers of sn/snoRNA features per `gtf`/`gff3` the same?</i></summary>
```{r Are the numbers of sn/snoRNA features per gtf/gff3 the same?, results='hide', message=FALSE, warning=FALSE}
gff3_SGD %>% colnames()
gff3_Ens %>% colnames()
gtf_Ens %>% colnames()

type_gff3_SGD <- gff3_SGD %>%
    dplyr::group_by(type) %>%
    dplyr::summarize(n = dplyr::n())

type_gff3_Ens <- gff3_Ens %>%
    dplyr::group_by(type) %>% 
    dplyr::summarize(n = dplyr::n())

type_gtf_Ens <- gtf_Ens %>%
    dplyr::group_by(type) %>% 
    dplyr::summarize(n = dplyr::n())

biotype_gff3_SGD <- gff3_SGD %>%
    dplyr::group_by(type) %>%
    dplyr::summarize(n = dplyr::n())

biotype_gff3_Ens <- gff3_Ens %>%
    dplyr::group_by(biotype) %>% 
    dplyr::summarize(n = dplyr::n())

biotype_gtf_Ens <- gtf_Ens %>%
    dplyr::group_by(gene_biotype) %>% 
    dplyr::summarize(n = dplyr::n())
```
</details>
<br />

### ...
#### Code
<details>
<summary><i>Code: ... </i></summary>
```{r}
#  Initialize function --------------------------------------------------------
`%notin%` <- Negate(`%in%`)


#  For the SGD gff3, subset by column "type" entry "gene" ---------------------
z_gff3_SGD_genes <- gff3_SGD %>%
    dplyr::filter(type == "gene")

#  Tally number of genes per chromosome
# z_gff3_SGD_genes %>%
#     dplyr::group_by(seqnames) %>%
#     dplyr::summarize(n = dplyr::n())

z_gff3_Ens_genes_summary <- gff3_Ens %>%
    dplyr::filter(type == "gene") %>%
    dplyr::group_by(seqnames) %>%
    dplyr::summarize(n = dplyr::n())

z_gff3_Ens_genes_summary_sum <- sum(z_gff3_Ens_genes_summary$n)


#  For the Ensembl gff3, subset by column "type" entry "gene" -----------------
z_gff3_Ens_genes <- gff3_Ens %>%
    dplyr::filter(type == "gene") %>%
    dplyr::group_by(seqnames)

z_gff3_SGD_genes$ID %notin%
    stringr::str_remove(z_gff3_Ens_genes$ID, "gene:") %>%
table()


#  Find gene IDs in SGD not in Ensembl ----------------------------------------
z_gff3_SGD_genes[z_gff3_SGD_genes$ID %notin%
        stringr::str_remove(z_gff3_Ens_genes$ID, "gene:"), ]$ID
# [1] "YER109C" "YFL057C" "YFL056C" "YOR031W" "R0010W"  "R0020C"  "R0030W"  "R0040C"

#  (column ID) ------------------------
c("YER109C", "YFL057C", "YFL056C", "YOR031W") %in%
    stringr::str_remove(gff3_Ens$ID, "gene:|CDS:|transcript:")
# [1] FALSE FALSE  TRUE FALSE

gff3_Ens[stringr::str_remove(gff3_Ens$ID, "gene:|CDS:|transcript:") %in%
    c("YER109C", "YFL057C", "YFL056C", "YOR031W"), ]

#  (column protein_ID) ----------------
c("YER109C", "YFL057C", "YFL056C", "YOR031W") %in% gff3_Ens$protein_id
# [1] FALSE FALSE FALSE FALSE

# gff3_Ens[gff3_Ens$protein_id %in%
#     c("YER109C", "YFL057C", "YFL056C", "YOR031W"), ]

#  (column gene_id) -------------------
c("YER109C", "YFL057C", "YFL056C", "YOR031W") %in% gff3_Ens$gene_id
# [1] FALSE FALSE TRUE FALSE

gff3_Ens[gff3_Ens$gene_id %in%
    c("YER109C", "YFL057C", "YFL056C", "YOR031W"), ]

#  (column transcript_id) -------------
c("YER109C", "YFL057C", "YFL056C", "YOR031W") %in% gff3_Ens$transcript_id
# [1] FALSE FALSE TRUE FALSE

gff3_Ens[gff3_Ens$transcript_id %in%
    c("YER109C", "YFL057C", "YFL056C", "YOR031W"), ]


#  Find any rows associated with "YER109C", "YFL057C", "YFL056C", "YOR031W" ---
library(data.table)
gff3_Ens[apply(gff3_Ens, 1, function(x) any(x %like% "YER109C")), ]  # 0 rows
gff3_Ens[apply(gff3_Ens, 1, function(x) any(x %like% "YFL057C")), ]  # 1 row (in row for "YFL056C")
gff3_Ens[apply(gff3_Ens, 1, function(x) any(x %like% "YFL056C")), ]  # 3 rows
gff3_Ens[apply(gff3_Ens, 1, function(x) any(x %like% "YOR031W")), ]  # 0 rows
```
</details>
<br />

"YER109C", "YFL057C", "YFL056C", and "YOR031W" are on, respectively, chromosomes V, VI, VI, and XV; "R0010W", "R0020C", "R0030W", and "R0040C" are on "2-micron". Thus, "YER109C", "YFL057C", "YFL056C", and "YOR031W" are apparently valid genes that we *may* care about that are not&mdash;for some reason&mdash;associated with IDs in the Ensembl annotation.

"YER109C", "YFL057C", "YFL056C", and "YOR031W" are not associated with `protein_ID` assignments in `gff3_Ens`; however, in `gff3_Ens`, "YFL056C" appears to be associated with `type` "pseudogene" and "pseudogene_transcript", `biotype` "pseudogene", and corresponding `gene_id` and `transcript_id` entries; the other genes&mdash;"YER109C", "YFL057C", and "YOR031W"&mdash;are not.

"YFL056C" is associated with three rows in `gff3_Ens`: `type`s "pseudogene", "pseudogenic_transcript", and "exon".

"YFL057C" is associated with the "YFL056C" `type` "pseudogene" row (it's in the `description` column)

### ...
#### Code
<details>
<summary><i>Code: ... </i></summary>
```{r}
#  repeat region --------------------------------------------------------------
z_gff3_SGD_repeat_region <- gff3_SGD %>%
    dplyr::filter(type == "repeat_region")

z_rr_y_NA <- z_gff3_SGD_repeat_region[is.na(z_gff3_SGD_repeat_region$ID), ]
z_rr_n_NA <- z_gff3_SGD_repeat_region[!is.na(z_gff3_SGD_repeat_region$ID), ]

z_rr_y_NA$Parent %>% unlist()
#  [1] "TEL02L-YP"  "TEL04R-YP"  "TEL05L-YP"  "TEL05R-YP"  "TEL06L-YP"  "TEL07R-YP"  "TEL08L-YP"  "TEL08R-YP"  "TEL09L-YP"  "TEL10L-YP"  "TEL12L-YP1" "TEL12L-YP2" "TEL12R-YP2" "TEL12R-YP1"
# [15] "TEL13L-YP"  "TEL14L-YP"  "TEL15R-YP"  "TEL16L-YP"  "TEL16R-YP"

z_rr_n_NA$Parent %>% unlist()
# character(0)


#  telomere -------------------------------------------------------------------
z_gff3_SGD_telomere <- gff3_SGD %>%
    dplyr::filter(type == "telomere")

z_tel_y_NA <- z_gff3_SGD_telomere[is.na(z_gff3_SGD_telomere$ID), ]
z_tel_n_NA <- z_gff3_SGD_telomere[!is.na(z_gff3_SGD_telomere$ID), ]

table(z_tel_n_NA$end %in% z_rr_n_NA$end)
# TRUE
#   32

table(z_tel_n_NA$start %in% z_rr_n_NA$start)
# TRUE
#   32

gr_tel_n_NA <- GenomicRanges::GRanges(z_tel_n_NA)
gr_rr_n_NA <- GenomicRanges::GRanges(z_rr_n_NA)

IRanges::countOverlaps(gr_tel_n_NA, gr_rr_n_NA)
tmp <- IRanges::findOverlaps(gr_tel_n_NA, gr_rr_n_NA) %>% tibble::as_tibble()
rm(tmp)

IRanges::countOverlaps(gr_rr_n_NA, gr_tel_n_NA)
tmp <- IRanges::findOverlaps(gr_rr_n_NA, gr_tel_n_NA) %>% tibble::as_tibble()
rm(tmp)


#  centromere -----------------------------------------------------------------
z_gff3_SGD_centromere <- gff3_SGD %>%
    dplyr::filter(type == "centromere")

gr_cen <- GenomicRanges::GRanges(z_gff3_SGD_centromere)

IRanges::countOverlaps(gr_cen, gr_tel_n_NA)
IRanges::countOverlaps(gr_cen, gr_rr_n_NA)

# rm(z_rr_n_NA, z_rr_y_NA, z_tel_n_NA, z_tel_y_NA)
# rm(z_gff3_SGD_repeat_region)
# rm(z_gff3_SGD_telomere)
```
</details>
<br />

For `type` "repeat_region", go with `!is.na()`, which encompasses the ranges in `is.na()`.

All of the ranges in `z_rr_n_NA` overlap the ranges in `z_tel_n_NA`. There are no overlaps between `z_gff3_SGD_centromere` and `z_rr_n_NA` (which might be expected), nor are there overlaps between `z_gff3_SGD_centromere` and `z_tel_n_NA` (which are not expected).

`#CONCLUSION` Thus, I think I should exclude `type` "repeat_region" in favor of `type` "telomere". "Centromere" can be included without any problems with any features, I think.

### ...
#### Code
<details>
<summary><i>Code: ... </i></summary>
```{r}
z_gff3_SGD_ncRNA <- gff3_SGD %>%
    dplyr::filter(type == "ncRNA")
z_gff3_SGD_noncoding_exon <- gff3_SGD %>%
    dplyr::filter(type == "noncoding_exon")
z_gff3_SGD_tRNA <- gff3_SGD %>%
    dplyr::filter(type == "tRNA")
z_gff3_SGD_snRNA_snoRNA <- gff3_SGD %>%
    dplyr::filter(type == "snRNA" | type == "snoRNA")
z_gff3_SGD_rRNA <- gff3_SGD %>%
    dplyr::filter(type == "rRNA")

nc_etc <- gff3_SGD %>%
    dplyr::filter(
        type == "ncRNA" |
        type == "noncoding_exon" |
        type == "tRNA" |
        type == "snoRNA" |
        type == "snRNA" |
        type == "rRNA"
    )

gr_noncoding_exon <- GenomicRanges::GRanges(z_gff3_SGD_noncoding_exon)
gr_noncoding_ncRNA <- GenomicRanges::GRanges(z_gff3_SGD_ncRNA)
gr_tRNA <- GenomicRanges::GRanges(z_gff3_SGD_tRNA)
gr_snRNA_snoRNA <- GenomicRanges::GRanges(z_gff3_SGD_snRNA_snoRNA)
gr_rRNA <- GenomicRanges::GRanges(z_gff3_SGD_rRNA)

sum_1 <- IRanges::countOverlaps(gr_noncoding_exon, gr_noncoding_ncRNA) %>% sum()
tmp_1 <- IRanges::findOverlaps(gr_noncoding_exon, gr_noncoding_ncRNA) %>%
    tibble::as_tibble()

sum_2 <- IRanges::countOverlaps(gr_noncoding_exon, gr_tRNA) %>% sum()
tmp_2 <- IRanges::findOverlaps(gr_noncoding_exon, gr_tRNA) %>%
    tibble::as_tibble()

sum_3 <- IRanges::countOverlaps(gr_noncoding_exon, gr_snRNA_snoRNA) %>% sum()
tmp_3 <- IRanges::findOverlaps(gr_noncoding_exon, gr_snRNA_snoRNA) %>%
    tibble::as_tibble()

sum_4 <- IRanges::countOverlaps(gr_noncoding_exon, gr_rRNA) %>% sum()
tmp_4 <- IRanges::findOverlaps(gr_noncoding_exon, gr_rRNA) %>%
    tibble::as_tibble()

z_gff3_SGD_noncoding_exon[z_gff3_SGD_noncoding_exon$Name %notin% z_gff3_SGD_noncoding_exon[c(tmp_1$queryHits, tmp_2$queryHits, tmp_3$queryHits), ]$Name, ]

nrow(z_gff3_SGD_noncoding_exon) - (sum_1 + sum_2 + sum_3)

z_gff3_SGD_noncoding_exon[tmp_1$queryHits, ]
z_gff3_SGD_noncoding_exon[tmp_2$queryHits, ]
z_gff3_SGD_noncoding_exon[tmp_3$queryHits, ]
z_gff3_SGD_noncoding_exon[tmp_4$queryHits, ]
```
</details>
<br />

`#CONCLUSION` We can exclude `type` "noncoding exon"&mdash;these will be included in "ncRNA", "tRNA", "snoRNA", "snRNA", "rRNA". I think they will not be included with anything else.
`#CONCLUSION` So, what to include (so far) for `type`:
- "centromere"
- "genes" (`#MAYBE` stratified by `orf_classification`)
- "ncRNA"
- "rRNA"
- "snRNA"
- "snoRNA"
- "telomere"
- "tRNA"

### ...
#### Code
<details>
<summary><i>Code: ... </i></summary>
```{r}
z_gff3_SGD_ARS <- gff3_SGD %>%
    dplyr::filter(type == "ARS")
z_gff3_SGD_LTR <- gff3_SGD %>%
    dplyr::filter(type == "long_terminal_repeat")
z_gff3_SGD_LTR_retrotransposon <- gff3_SGD %>%
    dplyr::filter(type == "LTR_retrotransposon")
z_gff3_SGD_TE_gene <- gff3_SGD %>%
    dplyr::filter(type == "transposable_element_gene")
```
</details>
<br />
