---
title: "work_normalization-etc_rough-draft"
author: "KA"
email: "kalavatt@fredhutch.org"
output: html_notebook
---

## Prepare data for DGE analyses
### Get situated
#### Set working directory
```{r, results='hide', message=FALSE}
p_local <- "/Users/kalavattam/Dropbox/FHCC"
p_wd <- "2022_transcriptome-construction/results/2023-0215"
setwd(paste(p_local, p_wd, sep = "/"))

rm(p_local, p_wd)
```


#### Load necessary libraries

```{r, results='hide', message=FALSE}
library(DESeq2)
library(GenomicRanges)
library(IRanges)
library(plyr)
library(tidyverse)
```


### Load in featureCounts table as tibble

```{r, results='hide', message=FALSE}
p_fc <- "outfiles_featureCounts/combined_SC_KL/UT_prim_UMI"
f_fc <- "UT_prim_UMI.featureCounts"
t_fc <- read.table(
    paste(p_fc, f_fc, sep = "/"), header = TRUE, row.names = 1
) %>% 
    tibble::rownames_to_column() %>%
    tibble::as_tibble()

rm(p_fc, f_fc)
```


### Clean up tibble column names

```{r, include=TRUE, results='hide', message=FALSE}
colnames(t_fc) <- colnames(t_fc) %>%
    gsub("rowname", "feature", .) %>%
    gsub("Chr", "chr", .) %>%
    gsub("Start", "start", .) %>%
    gsub("End", "end", .) %>%
    gsub("Strand", "strand", .) %>%
    gsub("Length", "length", .) %>%
    gsub("bams_renamed.UT_prim_UMI.", "", .) %>%
    gsub(".UT_prim_UMI.bam", "", .) %>%
    gsub("\\.d", "-d", .) %>%
    gsub("\\.n", "-n", .)
```


### Order tibble by chromosome names and feature start positions

```{r, results='hide', message=FALSE}
t_fc$chr <- t_fc$chr %>% as.factor()

chr_order <- c(
    "I", "II", "III", "IV", "V", "VI", "VII", "VIII", "IX", "X", "XI", "XII",
    "XIII", "XIV", "XV", "XVI", "Mito", "A", "B", "C", "D", "E", "F"
)
t_fc$chr <- ordered(t_fc$chr, levels = chr_order)

t_fc <- t_fc %>%
    dplyr::arrange(chr, start)
```


### Record tibble's positional information in GRanges object
(`pos_info` object will be used in the DESeq2 and subsequent steps)

```{r, results='hide', message=FALSE}
pos_info <- GenomicRanges::GRanges(
    seqnames = t_fc$chr,
    ranges = IRanges::IRanges(t_fc$start, t_fc$end),
    strand = t_fc$strand,
    length = t_fc$length,
    feature = t_fc$feature
)
```


## Run DGE analyses
### Establish variables for dds
dds is `DESeqDataSet` object

```{r}
samples <- colnames(t_fc)[7:length(colnames(t_fc))]
stringr::str_split(samples, "_")
```

This is an [R Markdown](http://rmarkdown.rstudio.com) Notebook. When you execute code within the notebook, the results appear beneath the code. 

Try executing this chunk by clicking the *Run* button within the chunk or by placing your cursor inside it and pressing *Cmd+Shift+Enter*. 

```{r}
plot(cars)
```

Add a new chunk by clicking the *Insert Chunk* button on the toolbar or by pressing *Cmd+Option+I*.
```{r}

```

When you save the notebook, an HTML file containing the code and output will be saved alongside it (click the *Preview* button or press *Cmd+Shift+K* to preview the HTML file). 

The preview shows you a rendered HTML copy of the contents of the editor. Consequently, unlike *Knit*, *Preview* does not run any R code chunks. Instead, the output of the chunk when it was last run in the editor is displayed.

