---
title: "work_assess-process_R64-1-1_gff3_part-2.Rmd"
author: "KA"
email: "kalavatt@fredhutch.org"
output:
    html_notebook:
        toc: yes
        toc_float: true
---

## Get situated
### Code
<details>
<summary><i>Code: Get situated</i></summary>

```{r Get situated, results='hide', message=FALSE, warning=FALSE}
#!/usr/bin/env Rscript

suppressMessages(library(GenomicRanges))
suppressMessages(library(IRanges))
suppressMessages(library(plyr))
suppressMessages(library(readxl))
suppressMessages(library(rtracklayer))
suppressMessages(library(tidyverse))

options(scipen = 999)
options(ggrepel.max.overlaps = Inf)

if(stringr::str_detect(getwd(), "kalavattam")) {
    p_local <- "/Users/kalavattam/Dropbox/FHCC"
} else {
    p_local <- "/Users/kalavatt/projects-etc"
}
p_wd <- "2022_transcriptome-construction/results/2023-0215"

setwd(paste(p_local, p_wd, sep = "/"))
getwd()

rm(p_local, p_wd)
```
</details>
<br />
<br />

## TBD 1
### Code: Load gtfs as tibbles
<details>
<summary><i>Code: Load gtfs as tibbles</i></summary>

```{r Load gtfs as tibbles, results='hide', message=FALSE, warning=FALSE}
#!/usr/bin/env Rscript

p_Q <- "outfiles_gtf-gff3/Trinity-GG/Q_N/filtered/locus"
f_Q <- "Q_mkc-4_gte-pctl-25.gtf"
gtf_Q <- rtracklayer::import(paste(p_Q, f_Q, sep = "/")) %>%
    tibble::as_tibble() %>%
    dplyr::select(-c(phase, score)) %>%
    dplyr::arrange(seqnames, start, strand)

p_G <- "outfiles_gtf-gff3/Trinity-GG/G_N/filtered/locus"
f_G <- "G1_mkc-4_gte-pctl-25.gtf"
gtf_G1 <- rtracklayer::import(paste(p_G, f_G, sep = "/")) %>%
    tibble::as_tibble() %>%
    dplyr::select(-c(phase, score)) %>%
    dplyr::arrange(seqnames, start, strand)

p_gtf_all <- "outfiles_gtf-gff3/comprehensive/S288C_reference_genome_R64-1-1_20110203"
f_gtf_all <- "processed_features-intergenic_sense-antisense.gtf"
gtf_all <- rtracklayer::import(paste(p_gtf_all, f_gtf_all, sep = "/")) %>%
    tibble::as_tibble() %>%
    dplyr::select(-c(phase, score)) %>%
    dplyr::arrange(seqnames, start, strand)

rm(p_Q, f_Q, p_G, f_G, p_gtf_all, f_gtf_all)
```
</details>
<br />
<br />

### Code: Evaluate overlaps between newly documented and official features
<details>
<summary><i>Code: Evaluate overlaps between newly documented and official features</i></summary>

```{r Evaluate overlaps between newly documented and official features, results='hide', message=FALSE, warning=FALSE}
#!/usr/bin/env Rscript

g_Q <- makeGRangesFromDataFrame(gtf_Q, keep.extra.columns = TRUE)
g_G1 <- makeGRangesFromDataFrame(gtf_G1, keep.extra.columns = TRUE)
g_all <- makeGRangesFromDataFrame(gtf_all, keep.extra.columns = TRUE)

overlap_Q_v_all <- IRanges::findOverlaps(g_Q, g_all)
overlap_G1_v_all <- IRanges::findOverlaps(g_G1, g_all)

run <- FALSE
if(base::isTRUE(run)) {
    test_1 <- gtf_Q[queryHits(overlap_Q_v_all), ]
    test_2 <- gtf_all[subjectHits(overlap_Q_v_all), ]
    
    run <- FALSE
    if(base::isTRUE(run)) rm(list = ls(pattern = "test_"))
}
```

```{r}
#PICKUPHERE #LATER
#  Create rows for features in "Q" overlapping features in "all"
wrt_Q_all <- dplyr::bind_cols(
    gtf_Q[queryHits(overlap_Q_v_all), ],
    gtf_all[subjectHits(overlap_Q_v_all), ]
) %>% dplyr::rename(
        seqnames = seqnames...1,
        start = start...2,
        end = end...3,
        width = width...4,
        strand = strand...5,
        source = source...6,
        type = type...7,
        gene_id = gene_id...9,
        transcript_id = transcript_id...10,
        seqnames_all = seqnames...14,
        start_all = start...15,
        end_all = end...16,
        width_all = width...17,
        strand_all = strand...18,
        source_all = source...19,
        type_all = type...20,
        gene_id_all = gene_id...21,
        transcript_id_all = transcript_id...22,
        category_all = type.1,
        orf_classification_all = orf_classification,
        source_id_all = source_id,
    )

#  Cast dataframe types
wrt_Q_all <- data.frame(
    lapply(wrt_Q_all, function(x) gsub("NA", NA_character_, x))
) %>%
    tibble::as_tibble()

run <- FALSE
if(base::isTRUE(run)) str(wrt_Q_all)

wrt_Q_all$start <- as.numeric(wrt_Q_all$start)
wrt_Q_all$end <- as.numeric(wrt_Q_all$end)
wrt_Q_all$width <- as.numeric(wrt_Q_all$width)
wrt_Q_all$start_all <- as.numeric(wrt_Q_all$start_all)
wrt_Q_all$end_all <- as.numeric(wrt_Q_all$end_all)
wrt_Q_all$width_all <- as.numeric(wrt_Q_all$width_all)

run <- TRUE
if(base::isTRUE(run)) str(wrt_Q_all)
```

```{r}
#  If they overlap after stratifying for 'chr' and 'strand', then organize rows
#+ into groups
wrt_Q_all_group_chr_strand <- plyr::ddply(
    wrt_Q_all,
    c("seqnames", "strand"),
    function(x) { 
        #  Check if a record should be linked with the previous record
        y <- c(NA, x$end[-nrow(x)])
        z <- ifelse(is.na(y), 0, y)
        z <- cummax(z)
        z[is.na(y)] <- NA
        x$previous_end <- z
        
        return(x)
    }
)
wrt_Q_all_group_chr_strand <- wrt_Q_all_group_chr_strand %>%
    dplyr::relocate(c(start_all, end_all), .after = end)
wrt_Q_all_group_chr_strand$new_group <-
    is.na(wrt_Q_all_group_chr_strand$previous_end) | 
        (
            wrt_Q_all_group_chr_strand$start >=
            wrt_Q_all_group_chr_strand$previous_end
        )
wrt_Q_all_group_chr_strand$group <- cumsum(wrt_Q_all_group_chr_strand$new_group)
wrt_Q_all_group_chr_strand <- wrt_Q_all_group_chr_strand %>%
    dplyr::mutate(type_id_all = paste0(category_all, ": ", gene_id_all))

#  Consider breaking genes down by ORF classification? However, some 128 "gene"
#+ annotations have NA ORF classifications
run <- FALSE
if(base::isTRUE(run)) {
    test <- wrt_Q_all_group_chr_strand[
        wrt_Q_all_group_chr_strand$category_all == "gene", 
    ]
    test$orf_classification_all %>% is.na() %>% table()
    
    run <- TRUE
    if(base::isTRUE(run)) rm(test)
}
```

```{r}
#  Calculate percent overlaps between "Q" and "all" features and vice versa
percent_overlap <- function(x_start, x_end, y_start, y_end){
    x_length <- abs((x_end + 1) - x_start)
    
    #  Determine "largest" start
    max_start <- max(c(
        x_start, y_start
    ))
    
    #  Determine "smallest" end
    min_end <- min(c(
        (x_end + 1), (y_end + 1)
    ))
    
    overlap <- ifelse(
        (min_end - max_start) <= 0, 0, (min_end - max_start)
    )
    
    percent_overlap <- ((overlap / x_length) * 100)
    
    return(percent_overlap)
}


test <- wrt_Q_all_group_chr_strand
test$pct_Q_over_all <- mapply(
    percent_overlap,
    test$start,
    test$end,
    test$start_all,
    test$end_all
)
test$pct_all_over_Q <- mapply(
    percent_overlap,
    test$start_all,
    test$end_all,
    test$start,
    test$end
)
test <- test %>%
    dplyr::relocate(
        c(pct_Q_over_all, pct_all_over_Q, type_id_all, group), .after = end_all
    )

# test_overlaps <- percent_overlap(
#     test$start, test$end, test$start_all, test$end_all
# )
# test_overlaps %>% table()
# 
# (test$start <= test$end_all) %>% table()
# (test$end >= test$start_all) %>% table()
# test_olap <- (test$start <= test$end_all) & (test$end >= test$start_all)


#  Aggregate the data
wrt_Q_all_group_chr_strand %>% colnames()
wrt_Q_all_group_chr_strand$type_id_all %>% head()
wrt_Q_all_agg_chr_strand <- plyr::ddply(
    wrt_Q_all_group_chr_strand,
    .(seqnames, strand, group),
    plyr::summarize, 
    start = min(start),
    end = max(end),
    width = (end - start) + 1,
    id = paste0(id, collapse = "; "),
    category = , #TODO #PICKUPHERE #TOMORROW
    name = paste0(type_id_all, collapse = "; ")
) %>%
    dplyr::mutate(
        strand = strand,
        source = "SGD (KA)",
        type = ifelse(
            Alias == "pseudogene" | Alias == "pseudogene; pseudogene",
            "PG",
            "TE"
        ),
        orf_classification = NA_character_,
        gene = NA_character_
    ) %>%
    dplyr::select(-group) %>% 
    dplyr::arrange(seqnames, start, strand) %>%
    dplyr::relocate(c(
        seqnames, start, end, width, strand, source, type, ID, dbxref, Name,
        Note, Ontology_term, orf_classification, gene
    ))
```

```{r}
details_Alias <- wrt_Q_all_agg_chr_strand$Alias %>%
    stringr::str_remove_all("s") %>%
    stringr::str_split("; ") %>%
    purrr::map(sort) %>%
    purrr::map(unique)
details_Alias <- sapply(
    details_Alias[!sapply(details_Alias, purrr::is_empty)],
    paste,
    collapse = " "
) %>%
    gsub("long_terminal_repeat", "LTR", .) %>%
    gsub("LTR_retrotranpoon", "RT", .) %>%
    gsub("tranpoable_element_gene", "TE", .) %>%
    gsub("peudogene", "PG", .)

details_Name <- wrt_Q_all_agg_chr_strand$Name %>%
    stringr::str_remove_all("s") %>%
    stringr::str_split("; ") %>%
    purrr::map(sort) %>%
    purrr::map(unique)
details_Name <- sapply(
    details_Name[!sapply(details_Name, purrr::is_empty)],
    paste,
    collapse = "; "
) %>%
    gsub("long_terminal_repeat", "LTR", .) %>%
    gsub("LTR_retrotranpoon", "RT", .) %>%
    gsub("tranpoable_element_gene", "TE_gene", .) %>%
    gsub("peudogene", "PG", .)

wrt_Q_all_agg_chr_strand$Name <- details_Name
wrt_Q_all_agg_chr_strand$Alias <- details_Alias
wrt_Q_all_agg_chr_strand$n_types <- ifelse(
    grepl(" ", wrt_Q_all_agg_chr_strand$Alias, fixed = TRUE),
    stringr::str_count(wrt_Q_all_agg_chr_strand$Alias, " ") + 1,
    1
)
wrt_Q_all_agg_chr_strand$n_features <- ifelse(
    grepl(";", wrt_Q_all_agg_chr_strand$ID, fixed = TRUE),
    stringr::str_count(wrt_Q_all_agg_chr_strand$ID, ";") + 1,
    1
)

wrt_Q_all <- wrt_Q_all_agg_chr_strand
wrt_Q_all$ID <- gsub("; ", "_", wrt_Q_all$ID)
wrt_Q_all$Ontology_term <- wrt_Q_all$Ontology_term %>%
    gsub("; ", "_", .) %>%
    gsub(", ", "-", .)
wrt_Q_all$Name <- wrt_Q_all$Name %>%
    gsub(": ", "-", .) %>%
    gsub("; ", "_", .) %>%
    gsub("TE_gene", "TE", .)
# wrt_Q_all %>%
#     dplyr::group_by(strand) %>%
#     dplyr::summarize(n = dplyr::n())

rm(wrt_Q_all_agg_chr_strand, wrt_Q_all_group_chr_strand, details_Alias, details_Name)

```
</details>
<br />
