---
title: "work_assessment-processing_gtfs"
author: "KA"
email: "kalavatt@fredhutch.org"
output:
    html_notebook:
        toc: yes
        toc_float: true
---
<br />

## Get started
### Get situated
#### Code
<details>
<summary><i>Code: Get situated</i></summary>
```{r}
#!/usr/bin/env Rscript

library(GenomicRanges)
library(IRanges)
library(readxl)
library(rtracklayer)
library(tidyverse)

options(scipen = 999)
options(ggrepel.max.overlaps = Inf)

if(stringr::str_detect(getwd(), "kalavattam")) {
    p_local <- "/Users/kalavattam/Dropbox/FHCC"
} else {
    p_local <- "/Users/kalavatt/projects-etc"
}
p_wd <- "2022_transcriptome-construction/results/2023-0215"

setwd(paste(p_local, p_wd, sep = "/"))
getwd()

rm(p_local, p_wd)
```
</details>
<br />
<br />

## Assess the non-`Trinity`-GG `gtf`/`gff3` files
### Describe the provenance of the files
#### Text
<details>
<summary><i>Text: Describe the provenance of the files</i></summary>

In notebook `work_assessment-processing_gtfs.md`, copied the following files into the new directory `infiles_gtf-gff3/already` within `2022_transcriptome-construction/results/2023-0215`:

- `combined_SC_KL_20S.gff3`
- `combined_SC_KL.gff3`
- `combined.gtf`
- `Kluyveromyces_lactis_gca_000002515.ASM251v1.55.gff3`
- `Saccharomyces_cerevisiae.R64-1-1.108.gtf`
- `Saccharomyces_cerevisiae.R64-1-1.108.plus-chr-rename.gtf`

For details regarding `combined_SC_KL.gff3`, including it creation, see `results/2022-1025/notebook.md` and `results/2022-1201/work-Trinity-1.md` (contains code for the creation of the `gff3` file).

Regarding `combined_SC_KL_20S.gff3`, see the above two files as well as `work_gff3_include-20S.md` (which will likely be merged with the contents of this notebook `#TODO` `#MAYBE` `#LATER`).

`combined.gtf` was created by someone who was in Christine Cuccinota's graduate laboratory&mdash;I don't really know if the material in `combined.gtf` is accurate (i.e., not missing anything, not containing errors from editing, etc.), let alone if it has been edited since its creation, the time of which I also don't know. But Alison Greenlaw seems fine using it. Certainly, the structure does not follow the WashU/UCSC data guidelines for `gtf` and related files...

`Kluyveromyces_lactis_gca_000002515.ASM251v1.55.gff3`:

- genome build: `Genolevures Consortium ASM251v1`
- genome version: `ASM251v1`
- genome build accession: `GCA_000002515.1`
- gene build last updated: `2015-02`
- *No editing/other changes*

`Saccharomyces_cerevisiae.R64-1-1.108.gtf`:

- genome build: `R64-1-1`
- genome version: `R64-1-1`
- genome date: `2011-09`
- genome build accession: `GCA_000146045.2`
- gene build last updated: `2018-10`
- *No editing/other changes*

`Saccharomyces_cerevisiae.R64-1-1.108.plus-chr-rename.gtf`:

- *Same as above*
- Editing performed/detailed in `results/2022-1101/work-TPM-calculation.md`

`#TODO` *Write up details for the Trinity-GG `gff3` files*
</details>
<br />

### What are the non-`Trinity-GG` `gtf`s/`gff3`s to assess?
#### Code
<details>
<summary><i>Code: What are the non-`Trinity-GG` `gtf`s/`gff3`s to assess?</i></summary>
```{r}
#!/usr/bin/env Rscript

path_g <- "infiles_gtf-gff3/already"
list.files(path_g)
```
</details>
<br />

### Examine data structures for read-in gtfs/gff3s
#### Load `gtf`s/`gff3`s
##### Code
<details>
<summary><i>Code: Load `gtf`s/`gff3`s</i></summary>

```{r}
#!/usr/bin/env Rscript

 g_sk2 <- rtracklayer::import(paste(path_g, list.files(path_g)[1], sep = "/"))
  g_sk <- rtracklayer::import(paste(path_g, list.files(path_g)[2], sep = "/"))
g_comb <- rtracklayer::import(paste(path_g, list.files(path_g)[3], sep = "/"))
   g_k <- rtracklayer::import(paste(path_g, list.files(path_g)[4], sep = "/"))
   g_s <- rtracklayer::import(paste(path_g, list.files(path_g)[5], sep = "/"))
 g_sre <- rtracklayer::import(paste(path_g, list.files(path_g)[6], sep = "/"))
```
</details>
<br />

#### Reviewing `GenomicRanges` accessor functions, etc.
Building on the material [here](https://bioconductor.org/packages/devel/bioc/vignettes/GenomicRanges/inst/doc/GenomicRangesIntroduction.html).

##### Code
<details>
<summary><i>Code: Reviewing `GenomicRanges` accessor functions, etc.</i></summary>

```{r}
#!/usr/bin/env Rscript

cat("# =====================================\n")
cat("# Review the GenomicRanges accessor functions by looking at Kluyveromyces_lactis_gca_000002515.ASM251v1.55.gff3")
cat("\n\n")

cat("# -------------------------------------\n")
cat("# seqnames()\n")
seqnames(g_k)
cat("\n\n")

cat("# -------------------------------------\n")
cat("# ranges()\n")
ranges(g_k)
cat("\n\n")

cat("# -------------------------------------\n")
cat("# strand()\n")
strand(g_k)
cat("\n\n")

cat("# -------------------------------------\n")
cat("# granges()\n")
granges(g_k)
cat("\n\n")

cat("# -------------------------------------\n")
cat("# mcols()\n")
mcols(g_k)
cat("\n\n")

cat("# -------------------------------------\n")
cat("# mcols() %>% colnames()\n")
#  Get all "columns" in GRanges object
mcols(g_k) %>% colnames()
cat("\n\n")

cat("# -------------------------------------\n")
cat("# seqlengths()\n")
cat("# (Apparently, will need to add this info yourself...)\n")
seqlengths(g_k)

cat("\n\n")

# g_s
```
</details>
<br />

#### Review the structure of a `GRanges` object
##### Code
<details>
<summary><i>Code: Review the structure of a `GRanges` object</i></summary>

```{r}
#!/usr/bin/env Rscript

cat("# =====================================\n")
cat("# Review the structure of a GRanges object with Kluyveromyces_lactis_gca_000002515.ASM251v1.55.gff3")
cat("\n\n")

cat("# -------------------------------------\n")
cat("# g_k %>% as.data.frame() %>% length()\n")
g_k %>% as.data.frame() %>% length()
cat("\n\n")

cat("# -------------------------------------\n")
cat("# mcols(g_k) %>% length()\n")
mcols(g_k) %>% length()
cat("\n\n")

cat("# -------------------------------------\n")
cat("#QUESTION Why the discrepancy?\n\n")

cat("# -------------------------------------\n")
cat("# g_k %>% as.data.frame() %>% colnames()\n")
g_k %>% as.data.frame() %>% colnames()
cat("\n\n")

cat("# -------------------------------------\n")
cat("# mcols(g_k) %>% colnames()\n")
mcols(g_k) %>% colnames()
cat("\n\n")

cat("# -------------------------------------\n")
cat("#ANSWER mcols() is returning the metadata columns, not all\n")
```
</details>
<br />

#### Explore/test a `GRanges`-to-`data.frame`-to-`GRanges` conversion
Building on info [here](https://web.mit.edu/~r/current/arch/i386_linux26/lib/R/library/GenomicRanges/html/makeGRangesFromDataFrame.html).

##### Code
<details>
<summary><i>Code: Explore/test a `GRanges`-to-`data.frame`-to-`GRanges` conversion</i></summary>

```{r}
#!/usr/bin/env Rscript

cat("# -------------------------------------\n")
cat("#QUESTION If I convert, say, g_k to a data.frame, can I successfully\n")
cat("          convert it back to a GRanges object?\n\n")

d_k <- g_k %>% as.data.frame()

#  "Reconstitute" the GRanges object ("rg")
rg_k <- GenomicRanges::makeGRangesFromDataFrame(
    d_k, keep.extra.columns = TRUE
)
isTRUE(length(mcols(g_k)) == length(mcols(rg_k)))

#  Eyeball the reconstituted files
rtracklayer::export(rg_k, "test.gtf")
rtracklayer::export(rg_k, "test.gff3")

#  It works! Empty vector elements are now included in the metadata of
#+ test.gff3 vs. Kluyveromyces_lactis_gca_000002515.ASM251v1.55.gff3; besides
#+ that, they are the same. Now, can delete the above test files.
unlink(c("test.gtf", "test.gff3"))
# file.exists(c("test.gtf", "test.gff3"))

cat("\n# -------------------------------------\n")
cat("#ANSWER Yes")
cat("\n\n")
```
</details>
<br />

## Describe details/game plan for the experiment
### The contents of `infiles_gtf-gff3/Trinity-GG`
#### Code
<details>
<summary><i>Code: The contents of `infiles_gtf-gff3/Trinity-GG`</i></summary>

```{bash}
#!/bin/bash

cd infiles_gtf-gff3/Trinity-GG
.,s 
```
</details>
<br />

#### Printed
<details>
<summary><i>Printed: The contents of `infiles_gtf-gff3/Trinity-GG`</i></summary>

```{txt}
./G_N:
total 43M
drwx------ 8 kalavatt  256 Feb 24 15:16 ./
drwx------ 4 kalavatt  128 Mar 27 11:08 ../
-rwx------ 1 kalavatt 6.8M Feb 24 15:07 trinity-gg_mkc-16_mir-0.05_mg-2_gf-0.05.gff3*
-rwx------ 1 kalavatt 8.2M Feb 24 15:10 trinity-gg_mkc-1_mir-0.05_mg-2_gf-0.05.gff3*
-rwx------ 1 kalavatt 7.6M Feb 24 15:09 trinity-gg_mkc-2_mir-0.05_mg-2_gf-0.05.gff3*
-rwx------ 1 kalavatt 6.5M Feb 24 15:07 trinity-gg_mkc-32_mir-0.05_mg-2_gf-0.05.gff3*
-rwx------ 1 kalavatt 7.0M Feb 24 15:09 trinity-gg_mkc-4_mir-0.05_mg-2_gf-0.05.gff3*
-rwx------ 1 kalavatt 6.8M Feb 24 15:08 trinity-gg_mkc-8_mir-0.05_mg-2_gf-0.05.gff3*

./Q_N:
total 52M
drwx------ 8 kalavatt  256 Feb 24 15:16 ./
drwx------ 4 kalavatt  128 Mar 27 11:08 ../
-rwx------ 1 kalavatt 8.3M Feb 24 14:44 trinity-gg_mkc-16_mir-0.05_mg-2_gf-0.05.gff3*
-rwx------ 1 kalavatt  11M Feb 24 14:47 trinity-gg_mkc-1_mir-0.05_mg-2_gf-0.05.gff3*
-rwx------ 1 kalavatt 9.5M Feb 24 14:46 trinity-gg_mkc-2_mir-0.05_mg-2_gf-0.05.gff3*
-rwx------ 1 kalavatt 6.9M Feb 24 14:44 trinity-gg_mkc-32_mir-0.05_mg-2_gf-0.05.gff3*
-rwx------ 1 kalavatt 8.7M Feb 24 14:46 trinity-gg_mkc-4_mir-0.05_mg-2_gf-0.05.gff3*
-rwx------ 1 kalavatt 8.3M Feb 24 14:45 trinity-gg_mkc-8_mir-0.05_mg-2_gf-0.05.gff3*
```
</details>
<br />

### Written descriptions
#### Background
##### Text
<details>
<summary><i>Text: Background</i></summary>

Details for how these files were made are in the following notebooks:

- The `fasta` files from running `Trinity` (genome-guided, or "GG," mode): `results/2023-0111/work_Trinity-GF-GG-optimization_submit-jobs.md`
- Aligning the `fasta` sequences to generate `gff3` files: `results/2023-0111/work_GMAP_rough-draft.md`
</details>
<br />

#### Experiment gameplan
##### Text
<details>
<summary><i>Text: Experiment gameplan</i></summary>

1. Use the `Perl` command-line program `AGAT` to convert the `gff3` files to `gtf` files (`agat_convert_sp_gff2gtf.pl`): e.g., `agat_convert_sp_gff2gtf.pl --gff "${stem}.gff" -o "${stem}.gtf"`
2. Use `GffCompare` to discard "matching and 'contained' [putative transcripts/transcript fragments, a.k.a. 'transfrags'] ... (i.e., collapse intron-redundant transfrags across all query files):" e.g., `gffcompare -C "${stem}.gtf" -o "${stem}.collapsed.gtf"`)
3. Use ~~`featureCounts`~~ `htseq-count` to perform feature counting with the converted, collapsed gtf files
4. Load resulting counts matrices and `gtf` files into `R`/`RStudio`; perform processing as necessary (using mix of `bioconductor` and `tidyverse` programs)
5. In terms of (raw, non-normalized) alignment-to-putative transcript/transcript fragment tallies, retain putative transcripts/transcript fragments (i.e., transfrags) with tallies greater than or equal to the...
    i. 5th percentile
    ii. 10th percentile
    iii. 15th percentile
    iv. 20th percentile
    v. 25th percentile
    vi. 30th percentile
    vii. 35th percentile
    viii. 40th percentile
6. Use `rtracklayer` to write the converted, collapsed, counts-filtered data objects as `gtf` files
7. Send the processed `gtf` files to Alison Greenlaw for visual inspection with `IGV`

For both Q and G1, Alison asked me to filter the assemblies generated with `Trinity` parameter `mkc` = 1, 4, 8. However, for the sake of completeness, and to understand clearly what "incorrectness" looks like, I will filter `mkc` = 2, 16, 32 as well. Remember that the `Trinity` parameters `mir`, `mg`, and `gf` were held at default levels for the specific assemblies to be eyeballed by Alison.
</details>
<br />
<br />

## 0&ndash;4. Command-line work
See [`#work_assessment-processing_gtfs.md`](./work_assessment-processing_gtfs.md), where *steps 0 through 4* are detailed. Below, we pick up with *step 5*.
<br />
<br />

## 5. Perform quantile filtering of putative transcripts/transcript fragments
### Get situated
#### Code
<details>
<summary><i>Code: Get situated</i></summary>

```{r}
#!/usr/bin/env Rscript

library(GenomicRanges)
library(IRanges)
library(readxl)
library(rtracklayer)
library(tidyverse)

options(scipen = 999)
options(ggrepel.max.overlaps = Inf)

if(stringr::str_detect(getwd(), "kalavattam")) {
    p_local <- "/Users/kalavattam/Dropbox/FHCC"
} else {
    p_local <- "/Users/kalavatt/projects-etc"
}
p_wd <- "2022_transcriptome-construction/results/2023-0215"

setwd(paste(p_local, p_wd, sep = "/"))
# getwd()

rm(p_local, p_wd)
```
</details>
<br />

### Read in and process `gtf` files (as `data.frame`s)
#### Code
<details>
<summary><i>Code: Read in and process `gtf` files (as `data.frame`s)</i></summary>

```{r}
#!/usr/bin/env Rscript

import_g <- function(file) {
    rtracklayer::import(file)
}


read_in_gtfs <- function(vector_files) {
    list <- sapply(
        vector_files,
        import_g,
        simplify = FALSE,
        USE.NAMES = TRUE
    )
    names(list) <- paste0("k", c(1, 16, 2, 32, 4, 8))
    
    df <- sapply(
        list,
        as.data.frame,
        simplify = FALSE,
        USE.NAMES = TRUE
    )
    return(df)
}


#  G1 ---------------------------------
path_gtf_G <- "outfiles_gtf-gff3/Trinity-GG/G_N"
files_G <- list.files(
    # path_gtf_G, pattern = "*.combined.sans-chr.gtf", full.names = TRUE  # Approach #1
    path_gtf_G, pattern = "*.gffread.gff3", full.names = TRUE  # Approach #2
)
l_G_gtf <- read_in_gtfs(files_G)


#  Q ----------------------------------
path_gtf_Q <- "outfiles_gtf-gff3/Trinity-GG/Q_N"
files_Q <- list.files(
    # path_gtf_Q, pattern = "*.combined.sans-chr.gtf", full.names = TRUE  # Approach #1
    path_gtf_Q, pattern = "*.gffread.gff3", full.names = TRUE  # Approach #2
)
l_Q_gtf <- read_in_gtfs(files_Q)


#  Comparisons ------------------------
# l_G_gtf[["k1"]] %>% head()
# l_G_gtf[["k2"]] %>% head()
# 
# l_G_gtf[["k1"]] %>% head()
# l_Q_gtf[["k1"]] %>% head()
# 
# l_Q_gtf[["k1"]] %>% head()
# l_Q_gtf[["k2"]] %>% head()
```
</details>
<br />

### Read in and process counts matrices
#### Code
<details>
<summary><i>Code: Read in and process counts matrices</i></summary>

```{r}
#!/usr/bin/env Rscript

read_in_mat <- function(vector_of_files) {
    out_list <- sapply(
        vector_of_files,
        readr::read_tsv,
        simplify = FALSE,
        USE.NAMES = TRUE
    )
    names(out_list) <- paste0("k", c(1, 16, 2, 32, 4, 8))
    
    return(out_list)
}


#  G1 ---------------------------------
path_mat_G <- "outfiles_htseq-count/Trinity-GG/G_N"

# files_G_sam <- list.files(  # Approach #1
#     path_mat_G, pattern = "*.yes.tsv", full.names = TRUE
# )
# l_G_mat_sam <- read_in_mat(files_G_sam) %>% suppressMessages()
# 
# 
# files_G_rev <- list.files(  # Approach #1
#     path_mat_G, pattern = "*.reverse.tsv", full.names = TRUE
# )
# l_G_mat_rev <- read_in_mat(files_G_rev) %>% suppressMessages()

files_G_locus <- list.files(  # Approach #2
    path_mat_G, pattern = "*-locus.tsv", full.names = TRUE
)
l_G_mat_locus <- read_in_mat(files_G_locus) %>% suppressMessages()


# files_G_mRNA <- list.files(  # Approach #2
#     path_mat_G, pattern = "*-mRNA.tsv", full.names = TRUE
# )
# l_G_mat_mRNA <- read_in_mat(files_G_mRNA) %>% suppressMessages()


#  Q ----------------------------------
path_mat_Q <- "outfiles_htseq-count/Trinity-GG/Q_N"

# files_Q_sam <- list.files(  # Approach #1
#     path_mat_Q, pattern = "*.yes.tsv", full.names = TRUE
# )
# l_Q_mat_sam <- read_in_mat(files_Q_sam) %>% suppressMessages()
# 
# files_Q_rev <- list.files(  # Approach #1
#     path_mat_Q, pattern = "*.reverse.tsv", full.names = TRUE
# )
# l_Q_mat_rev <- read_in_mat(files_Q_rev) %>% suppressMessages()

files_Q_locus <- list.files(  # Approach #2
    path_mat_Q, pattern = "*-locus.tsv", full.names = TRUE
)
l_Q_mat_locus <- read_in_mat(files_Q_locus) %>% suppressMessages()

# files_Q_mRNA <- list.files(  # Approach #2
#     path_mat_Q, pattern = "*-mRNA.tsv", full.names = TRUE
# )
# l_Q_mat_mRNA <- read_in_mat(files_Q_mRNA) %>% suppressMessages()


#  Comparisons ------------------------
# l_Q_mat_sam[["k1"]] %>% head()
# l_Q_mat_sam[["k2"]] %>% head()
# 
# l_Q_mat_sam[["k1"]] %>% head()
# l_Q_mat_rev[["k1"]] %>% head()


#  Rename columns ---------------------
#+ ...in each dataframe composing each list
rename_columns <- function(list) {
    lapply(
        list,
        function(df) {
            names(df) <- gsub("bams_renamed\\/UT_prim_UMI\\/WT_", "", names(df)) %>%
                gsub("_day1_ovn_", "_", .) %>%
                gsub("_day7_ovn_", "_", .) %>% 
                gsub("_aux-F_tc-F_", "_", .) %>%
                gsub("_tech1\\.UT_prim_UMI\\.bam", "", .) %>%
                gsub("^\\.\\.\\.1", "gene_id", .)
            
            return(df)
        }
    )
}

# l_G_mat_sam <- rename_columns(l_G_mat_sam)  # Approach #1
# l_G_mat_rev <- rename_columns(l_G_mat_rev)  # Approach #1
# l_Q_mat_sam <- rename_columns(l_Q_mat_sam)  # Approach #1
# l_Q_mat_rev <- rename_columns(l_Q_mat_rev)  # Approach #1

l_G_mat_locus <- rename_columns(l_G_mat_locus)  # Approach #2
# l_G_mat_mRNA <- rename_columns(l_G_mat_mRNA)  # Approach #2
l_Q_mat_locus <- rename_columns(l_Q_mat_locus)  # Approach #2
# l_Q_mat_mRNA <- rename_columns(l_Q_mat_mRNA)  # Approach #2

#  Check ------------------------------
# l_G_mat_sam$k32 %>% tail()
```
</details>
<br />

### Categorize putative transcripts/transcript fragments by percentile
#### Code
<details>
<summary><i>Code: Categorize putative transcripts/transcript fragments by percentile</i></summary>

```{r}
#!/usr/bin/env Rscript

filter_transfrags <- function(x, y) {
    df <- x
    
    if(y == "G1" | y == "G") {
        df <- x[, 1:3] %>% head(., -5)
    } else if(y == "Q") {
        df <- x[, c(1, 6, 7)] %>% head(., -5)
    } else if(y != "G1" | y != "G" | y != "Q") {
        stop("Parameter y must be either 'G1', 'G', or 'Q'")
    }
    
    df[, 4] <- df[, 2] + df[, 3]
    colnames(df)[4] <- "sum"

    ids <- list()

    ids$percentiles <- quantile(
        df$sum, probs = c(
            0.05, 0.10, 0.15, 0.20, 0.25, 0.30, 0.35, 0.40, 0.45, 0.50,
            0.55, 0.60, 0.65, 0.70, 0.75, 0.80, 0.85, 0.90, 0.95
        )
    ) %>%
        round()
    
    df$below_05 <- ifelse(df$sum <= ids$percentiles[1], TRUE, FALSE)
    df$below_10 <- ifelse(df$sum <= ids$percentiles[2], TRUE, FALSE)
    df$below_15 <- ifelse(df$sum <= ids$percentiles[3], TRUE, FALSE)
    df$below_20 <- ifelse(df$sum <= ids$percentiles[4], TRUE, FALSE)
    df$below_25 <- ifelse(df$sum <= ids$percentiles[5], TRUE, FALSE)
    df$below_30 <- ifelse(df$sum <= ids$percentiles[6], TRUE, FALSE)
    df$below_35 <- ifelse(df$sum <= ids$percentiles[7], TRUE, FALSE)
    df$below_40 <- ifelse(df$sum <= ids$percentiles[8], TRUE, FALSE)
    df$below_45 <- ifelse(df$sum <= ids$percentiles[9], TRUE, FALSE)
    df$below_50 <- ifelse(df$sum <= ids$percentiles[10], TRUE, FALSE)
    df$below_55 <- ifelse(df$sum <= ids$percentiles[11], TRUE, FALSE)
    df$below_60 <- ifelse(df$sum <= ids$percentiles[12], TRUE, FALSE)
    df$below_65 <- ifelse(df$sum <= ids$percentiles[13], TRUE, FALSE)
    df$below_70 <- ifelse(df$sum <= ids$percentiles[14], TRUE, FALSE)
    df$below_75 <- ifelse(df$sum <= ids$percentiles[15], TRUE, FALSE)
    df$below_80 <- ifelse(df$sum <= ids$percentiles[16], TRUE, FALSE)
    df$below_85 <- ifelse(df$sum <= ids$percentiles[17], TRUE, FALSE)
    df$below_90 <- ifelse(df$sum <= ids$percentiles[18], TRUE, FALSE)
    df$below_95 <- ifelse(df$sum <= ids$percentiles[19], TRUE, FALSE)
    
    ids$below_05 <- df[df$below_05 == FALSE, ]$gene_id
    ids$below_10 <- df[df$below_10 == FALSE, ]$gene_id
    ids$below_15 <- df[df$below_15 == FALSE, ]$gene_id
    ids$below_20 <- df[df$below_20 == FALSE, ]$gene_id
    ids$below_25 <- df[df$below_25 == FALSE, ]$gene_id
    ids$below_30 <- df[df$below_30 == FALSE, ]$gene_id
    ids$below_35 <- df[df$below_35 == FALSE, ]$gene_id
    ids$below_40 <- df[df$below_40 == FALSE, ]$gene_id
    ids$below_45 <- df[df$below_45 == FALSE, ]$gene_id
    ids$below_50 <- df[df$below_50 == FALSE, ]$gene_id
    ids$below_55 <- df[df$below_55 == FALSE, ]$gene_id
    ids$below_60 <- df[df$below_60 == FALSE, ]$gene_id
    ids$below_65 <- df[df$below_65 == FALSE, ]$gene_id
    ids$below_70 <- df[df$below_70 == FALSE, ]$gene_id
    ids$below_75 <- df[df$below_75 == FALSE, ]$gene_id
    ids$below_80 <- df[df$below_80 == FALSE, ]$gene_id
    ids$below_85 <- df[df$below_85 == FALSE, ]$gene_id
    ids$below_90 <- df[df$below_90 == FALSE, ]$gene_id
    ids$below_95 <- df[df$below_95 == FALSE, ]$gene_id
    
    return(ids)
    
    
    #  Checks -------------------------
    # df$below_05 %>% table()
    # df$below_10 %>% table()
    # df$below_15 %>% table()
    # df$below_20 %>% table()
    # df$below_25 %>% table()
    # df$below_30 %>% table()
    
    
    #  Scraps -----------------------------
    # quantile(
    #     test$G1_N_rep1, probs = c(0.05, 0.10, 0.15, 0.20, 0.25, 0.30)
    # ) %>%
    #     round()
    # quantile(
    #     test$G1_N_rep2, probs = c(0.05, 0.10, 0.15, 0.20, 0.25, 0.30)
    # ) %>%
    #     round()
}


# f_G_mat_sam <- sapply(  # Approach #1
#     l_G_mat_sam,
#     filter_transfrags,
#     simplify = FALSE,
#     USE.NAMES = TRUE,
#     y = "G1"
# )
# f_G_mat_rev <- sapply(  # Approach #1
#     l_G_mat_rev,
#     filter_transfrags,
#     simplify = FALSE,
#     USE.NAMES = TRUE,
#     y = "G1"
# )
# f_Q_mat_sam <- sapply(  # Approach #1
#     l_Q_mat_sam,
#     filter_transfrags,
#     simplify = FALSE,
#     USE.NAMES = TRUE,
#     y = "Q"
# )
# f_Q_mat_rev <- sapply(  # Approach #1
#     l_Q_mat_rev,
#     filter_transfrags,
#     simplify = FALSE,
#     USE.NAMES = TRUE,
#     y = "Q"
# )

f_G_mat_locus <- sapply(  # Approach #2
    l_G_mat_locus,
    filter_transfrags,
    simplify = FALSE,
    USE.NAMES = TRUE,
    y = "G1"
)
# f_G_mat_mRNA <- sapply(  # Approach #2
#     l_G_mat_mRNA,
#     filter_transfrags,
#     simplify = FALSE,
#     USE.NAMES = TRUE,
#     y = "G1"
# )
f_Q_mat_locus <- sapply(  # Approach #2
    l_Q_mat_locus,
    filter_transfrags,
    simplify = FALSE,
    USE.NAMES = TRUE,
    y = "Q"
)
# f_Q_mat_mRNA <- sapply(  # Approach #2
#     l_Q_mat_mRNA,
#     filter_transfrags,
#     simplify = FALSE,
#     USE.NAMES = TRUE,
#     y = "Q"
# )

#  Checks -------------------------
# f_G_mat_sam$k4$below_05 %>% length()
# f_G_mat_sam$k4$below_10 %>% length()
# f_G_mat_sam$k4$below_15 %>% length()
# f_G_mat_sam$k4$below_20 %>% length()
# f_G_mat_sam$k4$below_25 %>% length()
# f_G_mat_sam$k4$below_30 %>% length()
# 
# f_G_mat_rev$k4$below_05 %>% length()
# f_G_mat_rev$k4$below_10 %>% length()
# f_G_mat_rev$k4$below_15 %>% length()
# f_G_mat_rev$k4$below_20 %>% length()
# f_G_mat_rev$k4$below_25 %>% length()
# f_G_mat_rev$k4$below_30 %>% length()
# 
# f_Q_mat_sam$k4$below_05 %>% length()
# f_Q_mat_sam$k4$below_10 %>% length()
# f_Q_mat_sam$k4$below_15 %>% length()
# f_Q_mat_sam$k4$below_20 %>% length()
# f_Q_mat_sam$k4$below_25 %>% length()
# f_Q_mat_sam$k4$below_30 %>% length()
# 
# f_Q_mat_rev$k4$below_05 %>% length()
# f_Q_mat_rev$k4$below_10 %>% length()
# f_Q_mat_rev$k4$below_15 %>% length()
# f_Q_mat_rev$k4$below_20 %>% length()
# f_Q_mat_rev$k4$below_25 %>% length()
# f_Q_mat_rev$k4$below_30 %>% length()
# 
# f_G_mat_sam$k16$percentiles
# f_G_mat_rev$k16$percentiles
# 
# f_Q_mat_sam$k16$percentiles
# f_Q_mat_rev$k16$percentiles
```
</details>
<br />

### Subset `gtf` `data.frame` to retain percentile-filtered `gene_id`s
#### Code
<details>
<summary><i>Code: Subset `gtf` `data.frame` to retain percentile-filtered `gene_id`s</i></summary>

```{r}
#!/usr/bin/env Rscript

subset_dataframes <- function(x, y) {
    df <- list()
    mat <- list()
    out <- list()
    
    for(i in names(x)) {
        df[[i]] <- x[[i]]
        mat[[i]] <- y[[i]]

        #  Approach #1
        # out[[i]]$below_05 <- df[[i]][df[[i]]$gene_id %in% mat[[i]]$below_05, ]
        # out[[i]]$below_10 <- df[[i]][df[[i]]$gene_id %in% mat[[i]]$below_10, ]
        # out[[i]]$below_15 <- df[[i]][df[[i]]$gene_id %in% mat[[i]]$below_15, ]
        # out[[i]]$below_20 <- df[[i]][df[[i]]$gene_id %in% mat[[i]]$below_20, ]
        # out[[i]]$below_25 <- df[[i]][df[[i]]$gene_id %in% mat[[i]]$below_25, ]
        # out[[i]]$below_30 <- df[[i]][df[[i]]$gene_id %in% mat[[i]]$below_30, ]
        # out[[i]]$below_35 <- df[[i]][df[[i]]$gene_id %in% mat[[i]]$below_35, ]
        # out[[i]]$below_40 <- df[[i]][df[[i]]$gene_id %in% mat[[i]]$below_40, ]
        # out[[i]]$below_45 <- df[[i]][df[[i]]$gene_id %in% mat[[i]]$below_45, ]
        # out[[i]]$below_50 <- df[[i]][df[[i]]$gene_id %in% mat[[i]]$below_50, ]
        # out[[i]]$below_55 <- df[[i]][df[[i]]$gene_id %in% mat[[i]]$below_55, ]
        # out[[i]]$below_60 <- df[[i]][df[[i]]$gene_id %in% mat[[i]]$below_60, ]
        # out[[i]]$below_65 <- df[[i]][df[[i]]$gene_id %in% mat[[i]]$below_65, ]
        # out[[i]]$below_70 <- df[[i]][df[[i]]$gene_id %in% mat[[i]]$below_70, ]
        # out[[i]]$below_75 <- df[[i]][df[[i]]$gene_id %in% mat[[i]]$below_75, ]
        # out[[i]]$below_80 <- df[[i]][df[[i]]$gene_id %in% mat[[i]]$below_80, ]
        # out[[i]]$below_85 <- df[[i]][df[[i]]$gene_id %in% mat[[i]]$below_85, ]
        # out[[i]]$below_90 <- df[[i]][df[[i]]$gene_id %in% mat[[i]]$below_90, ]
        # out[[i]]$below_95 <- df[[i]][df[[i]]$gene_id %in% mat[[i]]$below_95, ]
        
        #  Approach #2
        out[[i]]$below_05 <- df[[i]][df[[i]]$ID %in% mat[[i]]$below_05, ]
        out[[i]]$below_10 <- df[[i]][df[[i]]$ID %in% mat[[i]]$below_10, ]
        out[[i]]$below_15 <- df[[i]][df[[i]]$ID %in% mat[[i]]$below_15, ]
        out[[i]]$below_20 <- df[[i]][df[[i]]$ID %in% mat[[i]]$below_20, ]
        out[[i]]$below_25 <- df[[i]][df[[i]]$ID %in% mat[[i]]$below_25, ]
        out[[i]]$below_30 <- df[[i]][df[[i]]$ID %in% mat[[i]]$below_30, ]
        out[[i]]$below_35 <- df[[i]][df[[i]]$ID %in% mat[[i]]$below_35, ]
        out[[i]]$below_40 <- df[[i]][df[[i]]$ID %in% mat[[i]]$below_40, ]
        out[[i]]$below_45 <- df[[i]][df[[i]]$ID %in% mat[[i]]$below_45, ]
        out[[i]]$below_50 <- df[[i]][df[[i]]$ID %in% mat[[i]]$below_50, ]
        out[[i]]$below_55 <- df[[i]][df[[i]]$ID %in% mat[[i]]$below_55, ]
        out[[i]]$below_60 <- df[[i]][df[[i]]$ID %in% mat[[i]]$below_60, ]
        out[[i]]$below_65 <- df[[i]][df[[i]]$ID %in% mat[[i]]$below_65, ]
        out[[i]]$below_70 <- df[[i]][df[[i]]$ID %in% mat[[i]]$below_70, ]
        out[[i]]$below_75 <- df[[i]][df[[i]]$ID %in% mat[[i]]$below_75, ]
        out[[i]]$below_80 <- df[[i]][df[[i]]$ID %in% mat[[i]]$below_80, ]
        out[[i]]$below_85 <- df[[i]][df[[i]]$ID %in% mat[[i]]$below_85, ]
        out[[i]]$below_90 <- df[[i]][df[[i]]$ID %in% mat[[i]]$below_90, ]
        out[[i]]$below_95 <- df[[i]][df[[i]]$ID %in% mat[[i]]$below_95, ]
    }
    
    return(out)
}


# out_G_gtf <- subset_dataframes(l_G_gtf, f_G_mat_sam)  # Approach #1
# out_Q_gtf <- subset_dataframes(l_Q_gtf, f_Q_mat_sam)  # Approach #1

out_G_gtf_locus <- subset_dataframes(l_G_gtf, f_G_mat_locus)  # Approach #2
# out_G_gtf_mRNA <- subset_dataframes(l_G_gtf, f_G_mat_mRNA)  # Approach #2
out_Q_gtf_locus <- subset_dataframes(l_Q_gtf, f_Q_mat_locus)  # Approach #2
# out_Q_gtf_mRNA <- subset_dataframes(l_Q_gtf, f_Q_mat_mRNA)  # Approach #2
```
</details>
<br />

### Write out processed `gtf`s
#### Code
<details>
<summary><i>Code: Write out processed `gtf`s</i></summary>

```{r}
#!/usr/bin/env Rscript

# path_G <- "outfiles_gtf-gff3/Trinity-GG/G_N/filtered"  # Approach #1
# if(isFALSE(dir.exists(path_G))) dir.create(path_G)
# 
# path_Q <- "outfiles_gtf-gff3/Trinity-GG/Q_N/filtered"  # Approach #1
# if(isFALSE(dir.exists(path_Q))) dir.create(path_Q)

path_G_locus <- "outfiles_gtf-gff3/Trinity-GG/G_N/filtered/locus"  # Approach #2
if(isFALSE(dir.exists(path_G_locus))) dir.create(path_G_locus, recursive = TRUE)

path_Q_locus <- "outfiles_gtf-gff3/Trinity-GG/Q_N/filtered/locus"  # Approach #2
if(isFALSE(dir.exists(path_Q_locus))) dir.create(path_Q_locus, recursive = TRUE)

#  Approach #1
# for(i in names(out_G_gtf)) {
#     for(j in c(paste0(
#         "below_",
#         c(
#             "05", "10", "15", "20", "25", "30", "35", "40", "45", "50",
#             "55", "60", "65", "70", "75", "80", "85", "90", "95"
#         )
#     ))) {
#         # i <- "k2"
#         # j <- "below_05"
#         cat("Saving G1", i, j, "\n")
#         cat(paste0(
#             path_G, "/G1_mkc-", gsub("k", "", i), "_",
#             gsub("below_", "gte-pctl-", j), ".gtf"
#         ))
#         cat("\n\n")
#         
#         out_G_conv <- GenomicRanges::makeGRangesFromDataFrame(
#             out_G_gtf[[i]][[j]], keep.extra.columns = TRUE
#         )
# 
#         rtracklayer::export(
#             out_G_gtf[[i]][[j]],
#             paste0(
#                 path_G, "/G1_mkc-", gsub("k", "", i), "_",
#                 gsub("below_", "gte-pctl-", j), ".gtf"
#             )
#         )
#         
#         cat("Saving Q", i, j, "\n")
#         cat(paste0(
#             path_Q, "/Q_mkc-", gsub("k", "", i), "_",
#             gsub("below_", "gte-pctl-", j), ".gtf"
#         ))
#         cat("\n\n")
#         
#         out_Q_conv <- GenomicRanges::makeGRangesFromDataFrame(
#             out_Q_gtf[[i]][[j]], keep.extra.columns = TRUE
#         )
# 
#         rtracklayer::export(
#             out_Q_gtf[[i]][[j]],
#             paste0(
#                 path_Q, "/Q_mkc-", gsub("k", "", i), "_",
#                 gsub("below_", "gte-pctl-", j), ".gtf"
#             )
#         )
#     }
# }

#  Approach #2
for(i in names(out_G_gtf_locus)) {
    for(j in c(paste0(
        "below_",
        c(
            "05", "10", "15", "20", "25", "30", "35", "40", "45", "50",
            "55", "60", "65", "70", "75", "80", "85", "90", "95"
        )
    ))) {
        # i <- "k2"
        # j <- "below_05"
        cat("Saving G1", i, j, "\n")
        cat(paste0(
            path_G_locus, "/G1_mkc-", gsub("k", "", i), "_",
            gsub("below_", "gte-pctl-", j), ".gtf"
        ))
        cat("\n\n")
        
        out_G_conv <- GenomicRanges::makeGRangesFromDataFrame(
            out_G_gtf_locus[[i]][[j]], keep.extra.columns = TRUE
        )

        rtracklayer::export(
            out_G_gtf_locus[[i]][[j]],
            paste0(
                path_G_locus, "/G1_mkc-", gsub("k", "", i), "_",
                gsub("below_", "gte-pctl-", j), ".gtf"
            )
        )
        
        cat("Saving Q", i, j, "\n")
        cat(paste0(
            path_Q_locus, "/Q_mkc-", gsub("k", "", i), "_",
            gsub("below_", "gte-pctl-", j), ".gtf"
        ))
        cat("\n\n")
        
        out_Q_conv <- GenomicRanges::makeGRangesFromDataFrame(
            out_Q_gtf_locus[[i]][[j]], keep.extra.columns = TRUE
        )

        rtracklayer::export(
            out_Q_gtf_locus[[i]][[j]],
            paste0(
                path_Q_locus, "/Q_mkc-", gsub("k", "", i), "_",
                gsub("below_", "gte-pctl-", j), ".gtf"
            )
        )
    }
}
```
</details>
<br />
