---
title: "work_normalization-etc_rough-draft"
author: "KA"
email: "kalavatt@fredhutch.org"
output: html_notebook
---

## Prepare data for DGE analyses
### Get situated
#### Set working directory
```{r, results='hide', message=FALSE}
p_local <- "/Users/kalavattam/Dropbox/FHCC"
p_wd <- "2022_transcriptome-construction/results/2023-0215"
setwd(paste(p_local, p_wd, sep = "/"))

rm(p_local, p_wd)
```


#### Load necessary libraries

```{r, results='hide', message=FALSE}
# library(DESeq2)
library(GenomicRanges)
library(IRanges)
library(tidyverse)
```


### Load in and process featureCounts table

```{r, results='hide', message=FALSE}
#  Load in featureCounts table ------------------------------------------------
p_fc <- "outfiles_featureCounts/combined_SC_KL/UT_prim_UMI"
f_fc <- "UT_prim_UMI.featureCounts"
t_fc <- read.table(
    paste(p_fc, f_fc, sep = "/"), header = TRUE, row.names = 1
) %>% 
    tibble::rownames_to_column() %>%
    tibble::as_tibble()

rm(p_fc, f_fc)


#  Clean up tibble column names -----------------------------------------------
colnames(t_fc) <- colnames(t_fc) %>%
    gsub("rowname", "feature_init", .) %>%
    gsub("Chr", "chr", .) %>%
    gsub("Start", "start", .) %>%
    gsub("End", "end", .) %>%
    gsub("Strand", "strand", .) %>%
    gsub("Length", "length", .) %>%
    gsub("bams_renamed\\.UT_prim_UMI\\.", "", .) %>%
    gsub(".UT_prim_UMI.bam", "", .) %>%
    gsub("\\.d", "-d", .) %>%
    gsub("\\.n", "-n", .)


#  Order tibble by chromosome names and feature start positions ---------------
chr_SC <- c(
    "I", "II", "III", "IV", "V", "VI", "VII", "VIII", "IX", "X", "XI", "XII",
    "XIII", "XIV", "XV", "XVI", "Mito"
)
chr_KL <- c("A", "B", "C", "D", "E", "F")
chr_order <- c(chr_SC, chr_KL)
t_fc$chr <- t_fc$chr %>% as.factor()
t_fc$chr <- ordered(t_fc$chr, levels = chr_order)

t_fc <- t_fc %>% dplyr::arrange(chr, start)


#  Categorize chromosomes by genome of origin ---------------------------------
t_fc$genome <- ifelse(
    t_fc$chr %in% chr_SC,
    "S_cerevisiae",
    ifelse(
        t_fc$chr %in% chr_KL,
        "K_lactis",
        NA
    )
) %>%
    as.factor()

t_fc <- t_fc %>% dplyr::relocate("genome", .before = "chr")

#  Check on variable "genome"
levels(t_fc$genome)
t_fc %>%
    dplyr::group_by(genome) %>%
    dplyr::summarize(tally = length(genome))
# K_lactis = 5659, S_cerevisiae = 7507

rm(chr_KL, chr_SC, chr_order)


#  Split and better organize variable 'feature_init' --------------------------
split_isolate_convert <- function(in_vector, field, column_name) {
    df <- in_vector %>%
        stringr::str_split(., c("_")) %>%
        sapply(., "[", field) %>%
        as.data.frame() %>%
        tibble::as_tibble()
    
    colnames(df) <- column_name
    
    return(df)
}


el_1 <- split_isolate_convert(
    in_vector = t_fc$feature_init,
    field = 1,
    column_name = "feature"
)
el_2 <- split_isolate_convert(
    in_vector = t_fc$feature_init,
    field = 2,
    column_name = "type"
)
el_2$type <- addNA(el_2$type)  # Convert to factor and include 'NA' as a level

#  Check on the split information
levels(el_2$type)  # 233 levels
el_2 %>%
    dplyr::group_by(type) %>%
    dplyr::summarize(tally = length(type))  # mRNA-E1 = 6600, NA = 5547, etc.

#  Append split information to tibble 't_fc'
t_fc <- dplyr::bind_cols(t_fc, el_1, el_2) %>%
    dplyr::relocate(c("feature", "type"), .after = "feature_init")
t_fc

rm(el_1, el_2)
```


### Record tibble's positional information in GRanges object
(`pos_info` object will be used in the DESeq2 and subsequent steps)

```{r, results='hide', message=FALSE}
pos_info <- GenomicRanges::GRanges(
    seqnames = t_fc$chr,
    ranges = IRanges::IRanges(t_fc$start, t_fc$end),
    strand = t_fc$strand,
    length = t_fc$length,
    feature = t_fc$feature
)
```


## Run DGE analyses
### Establish variables for dds
- dds is `DESeqDataSet` object
- variables for dds are
    + "strain"
    + "state"
    + "time"
    + "RNA-seq"
    + "replicate"

```{r}
#  Columns 7 through to the last column are samples; get them into a vector
samples <- colnames(t_fc)[7:length(colnames(t_fc))]

#  Convert vector to list by splitting each element at underscores
samples <- stringr::str_split(samples, "_")

#  Clean up the WT "CU" samples (elements 47 and 50) so that each element has
#+ the same number of entries
samples[[47]][5] <- paste(samples[[47]][5], samples[[47]][6], sep = "-")
samples[[50]][5] <- paste(samples[[50]][5], samples[[50]][6], sep = "-")
samples[[47]] <- samples[[47]][-6]
samples[[50]] <- samples[[50]][-6]

#  Convert the list to a df, transpose it, convert it to a tibble, then name
#+ the columns
samples <- samples %>%
    as.data.frame(
        .,
        col.names = c(seq(1, 55)),
        row.names = c("strain", "state", "time", "RNA-seq", "replicate")
    ) %>%
    t() %>%
    tibble::as_tibble()
samples
```

```{r}

```











