
`#emails-with-AG.md`

<details>
<summary><b><font size="+2"><i>Table of contents</i></font></b></summary>
<!-- MarkdownTOC -->

1. [Emails with Alison, etc.](#emails-with-alison-etc)
	1. [Email: *IGV Sharing*](#email-igv-sharing)
		1. [1: Alison → me `Wednesday, 2022-1123, 4:41 PM`](#1-alison-%E2%86%92-me-wednesday-2022-1123-441-pm)
	1. [Email: *Questions, etc. for transcriptome assembly*](#email-questions-etc-for-transcriptome-assembly)
		1. [1: Me → Alison `Friday, 2022-1202, 12:53`](#1-me-%E2%86%92-alison-friday-2022-1202-1253)
			1. [Files](#files)
		1. [2: Alison → Me `Thursday, 2022-1202, 13:32`](#2-alison-%E2%86%92-me-thursday-2022-1202-1332)
	1. [Email: *Notes on Trinity*](#email-notes-on-trinity)
		1. [1: Alison → Me `Friday, 2022-1202, 16:58`](#1-alison-%E2%86%92-me-friday-2022-1202-1658)
			1. [Files](#files-1)
		1. [2: Me → Alison `Friday, 2022-1202, 17:26`](#2-me-%E2%86%92-alison-friday-2022-1202-1726)
		1. [3: Alison → Me `Friday, 2022-1202, 17:49`](#3-alison-%E2%86%92-me-friday-2022-1202-1749)
			1. [Files](#files-2)
		1. [4: Me → Alison `Saturday, 2022-1203, 9:10`](#4-me-%E2%86%92-alison-saturday-2022-1203-910)
	1. [Email: *Blacklist file*](#email-blacklist-file)
		1. [1: Alison → me `Monday, 2022-1205, 3:45 PM`](#1-alison-%E2%86%92-me-monday-2022-1205-345-pm)
		1. [2: Me → Alison `Monday, 2022-1205, 4:05 PM`](#2-me-%E2%86%92-alison-monday-2022-1205-405-pm)
		1. [3: Me → Alison `Tuesday, 2022-1206, 11:10 AM`](#3-me-%E2%86%92-alison-tuesday-2022-1206-1110-am)
		1. [4: Me → Alison `Tuesday, 2022-1206, 11:16 AM`](#4-me-%E2%86%92-alison-tuesday-2022-1206-1116-am)
		1. [5: Alison → me `Tuesday, 2022-1206, 11:44 AM`](#5-alison-%E2%86%92-me-tuesday-2022-1206-1144-am)
			1. [*Note fr/Alison on what to blacklist and what to not*](#note-fralison-on-what-to-blacklist-and-what-to-not)
		1. [6: Me → Alison `Tuesday, 2022-1206, 11:47 AM`](#6-me-%E2%86%92-alison-tuesday-2022-1206-1147-am)
		1. [7: Me → Alison `Tuesday, 2022-1206, 3:34 PM`](#7-me-%E2%86%92-alison-tuesday-2022-1206-334-pm)
		1. [8: Alison → me `Wednesday, 2022-1207, 12:20 PM`](#8-alison-%E2%86%92-me-wednesday-2022-1207-1220-pm)
		1. [9: Me → Alison `Wednesday, 2022-1207, 12:31 PM`](#9-me-%E2%86%92-alison-wednesday-2022-1207-1231-pm)
		1. [10: Me → Alison `Wednesday, 2022-1207, 12:39 PM`](#10-me-%E2%86%92-alison-wednesday-2022-1207-1239-pm)
	1. [Email: *Bam files in directory `~/tsukiyamalab/alisong/Kris_bams/`*](#email-bam-files-in-directory-~tsukiyamalabalisongkris_bams)
		1. [1: Me → Alison `Monday, 2022-1212, 3:27 PM`](#1-me-%E2%86%92-alison-monday-2022-1212-327-pm)
		1. [2: Alison → Me `Monday, 2022-1212, 3:42 PM`](#2-alison-%E2%86%92-me-monday-2022-1212-342-pm)
	1. [Email: *Gff3 files in `~/tsukiyamalab/alisong/Kris/gtfs_2022-1214/param_gene-overlap_{FALSE,TRUE}`*](#email-gff3-files-in-~tsukiyamalabalisongkrisgtfs_2022-1214param_gene-overlap_falsetrue)
		1. [1: Me → Alison `Wednesday, 2022-1213, 3:17 PM`](#1-me-%E2%86%92-alison-wednesday-2022-1213-317-pm)
	1. [Email: *Preprocessing Thoughts*](#email-preprocessing-thoughts)
		1. [1: Alison → Me `Thursday, 2022-1215, 19:07`](#1-alison-%E2%86%92-me-thursday-2022-1215-1907)
			1. [Files](#files-3)
		1. [2: Me → Alison `Friday, 2022-1216, 12:07`](#2-me-%E2%86%92-alison-friday-2022-1216-1207)
	1. [Email: *split script*](#email-split-script)
		1. [1: Alison → Me `Friday, 2022-1216, 14:50`](#1-alison-%E2%86%92-me-friday-2022-1216-1450)
			1. [Files](#files-4)
	1. [Email: *More trinity thoughts*](#email-more-trinity-thoughts)
		1. [1: Alison → Me `Friday, 2022-1216, 18:13`](#1-alison-%E2%86%92-me-friday-2022-1216-1813)
			1. [Files](#files-5)
		1. [2: Me → Alison `Monday, 2022-1219, 11:49`](#2-me-%E2%86%92-alison-monday-2022-1219-1149)
	1. [Email: *Update*](#email-update)
		1. [1: Me → Alison `Monday, 2022-1221, 12:02`](#1-me-%E2%86%92-alison-monday-2022-1221-1202)
		1. [2: Alison → Me `Monday, 2022-1221, 12:07`](#2-alison-%E2%86%92-me-monday-2022-1221-1207)
		1. [3: Me → Alison `Monday, 2022-1221, 17:50`](#3-me-%E2%86%92-alison-monday-2022-1221-1750)
		1. [4: Me → Alison `Tuesday, 2022-1222, 11:57`](#4-me-%E2%86%92-alison-tuesday-2022-1222-1157)
		1. [5: Me → Alison `Tuesday, 2022-1222, 12:00`](#5-me-%E2%86%92-alison-tuesday-2022-1222-1200)
		1. [6: Alison → Me `Tuesday, 2022-1221, 12:44`](#6-alison-%E2%86%92-me-tuesday-2022-1221-1244)
		1. [7: Alison → Me `Tuesday, 2022-1222, 14:12`](#7-alison-%E2%86%92-me-tuesday-2022-1222-1412)
	1. [Email: *Preprocessing - more thoughts!*](#email-preprocessing---more-thoughts)
		1. [1: Alison → Me `Tuesday, 2022-1222, 16:55`](#1-alison-%E2%86%92-me-tuesday-2022-1222-1655)
			1. [Files](#files-6)

<!-- /MarkdownTOC -->
</details>
<br />

<a id="emails-with-alison-etc"></a>
## Emails with Alison, etc.
<a id="email-blacklist-file"></a>
<a id="email-igv-sharing"></a>
### Email: *IGV Sharing*
2022-1123
<details>
<summary><i>Click to view</i></summary>

<a id="1-alison-%E2%86%92-me-wednesday-2022-1123-441-pm"></a>
#### 1: Alison → me `Wednesday, 2022-1123, 4:41 PM`
Hi Kris - 

I have made a folder where we can easily create IGV sessions to share and compare annotation. 

It can be found at:  
`~/tsukiyamalab/alisong/Kris_IGV_Sharing`

If you go to IGV and then go file, open session, you can open the .xml file with the bigwigs and annotation files I have loaded. You could also save a new xml file with additional files, focused on a certain region or with a particular scaling. Please note all bigwigs/bams/annotation files that you want to save within a session need to be within the same folder for the .xml file to work. I made a simple one with just Q, but I imagine this folder filling up with stuff as we continue to work. Feel free to copy stuff into that folder for ease. I assume none of these files will be particularly large so duplicates should have a negligible effect on storage costs. 

Enjoy your Thanksgiving!

Alison
</details>
<br />

<a id="email-questions-etc-for-transcriptome-assembly"></a>
### Email: *Questions, etc. for transcriptome assembly*
2022-1202
<details>
<summary><i>Click to view</i></summary>

<a id="1-me-%E2%86%92-alison-friday-2022-1202-1253"></a>
#### 1: Me → Alison `Friday, 2022-1202, 12:53`
Hi Alison,

To expedite things for transcriptome assembly, I put together some questions, comments, and notes. I’ve copied them over from my lab notebook into a .txt file. I also included screen grabs that might be easier to read.

The most important thing, I think, is for me to get some kind of an itemized/categorized list of problems with the previous assembly (or assemblies) together with descriptions of the problem (i.e., what we observe), what we “want”/would expect in ideal situations, and—for each itemized/categorized problem—a visual example or two with genomic coordinates that I could reference in IGV. It’s a big ask, I know, but I think it would be immensely helpful—for my own education, for me to get on the same wavelength as you, and for my upcoming email to Brian Haas, the author/maintainer of Trinity. (Also, I checked with Toshi, and he is OK with discussing relevant aspects of the project with Brian.)

Please take your time with this—maybe you can work on it when you have non-benchwork-or-other-responsibility time.

If you don’t know the answer to any of these questions, or if the answers are simply ‘no,’ then no worries at all.

And then, of course, apologies for any spelling or grammatical errors in this stuff.

Thanks!  
Kris

<a id="files"></a>
##### Files
[File: `KA.2022-1202.questions-comments-notes.1.png`](./notebook/KA.2022-1202.questions-comments-notes.1.png)
[File: `KA.2022-1202.questions-comments-notes.2.png`](./notebook/KA.2022-1202.questions-comments-notes.2.png)
[File: `KA.2022-1202.questions-comments-notes.txt`](./notebook/KA.2022-1202.questions-comments-notes.txt)

<a id="2-alison-%E2%86%92-me-thursday-2022-1202-1332"></a>
#### 2: Alison → Me `Thursday, 2022-1202, 13:32`
Hi Kris - 

I am going to focus on the "#IMPORTANT" section with the hope of getting you a decent write up by end of day. If I have time, I will also begin to address the other questions. 

That said - here are some questions I can answer very quickly. 

> Is there a reason Alison aligned her data with Bowtie 2, a non-splice-aware aligner, versus something like HISAT2 or STAR, both splice-aware aligners?
- We had the pipeline working already is the answer. Also our yeast only contain a few hundred introns (283 out of 6600 genes). Here is the database I use with specific intron information: http://intron.ucsc.edu/yeast4.1/
- If you think changing alignment will meaningfully improve data quality, then I am open to the suggestion. It may be something reviewers insist on. I can see introns in IGV at annotated sites - but I guess it's possible we have been missing novel Q introns. 

>When evaluating nascent RNA expression, do we need to consider introns, splicing, etc. at all? Like, would we expect nascent or stready-state 4tU-seq signal to run into introns or skip them?
- The Nascent 4tU contains more pre-mRNA (not yet spliced) than the steady state. By eye, splicing is much slower in Q cells than in G1 cells, as pre-mRNA makes up a larger fraction of the transcript. This is visually apparent without a splice aware aligner but might be even more dramatic with STAR or HISAT2. Annotation should not focus on splice sites.

Alison 
</details>
<br />

<a id="email-notes-on-trinity"></a>
### Email: *Notes on Trinity*
2022-1202-1203
<details>
<summary><i>Click to view</i></summary>

<a id="1-alison-%E2%86%92-me-friday-2022-1202-1658"></a>
#### 1: Alison → Me `Friday, 2022-1202, 16:58`
Hi Kris - 

Please let me know if you need more information or clarification. Hopefully I have addressed a lot of your questions and concerns with these two documents. FYI the word document needs to be in "web layout" under the view tab to look correct. 

Have a great weekend!

Alison

<a id="files-1"></a>
##### Files
[File: `AG.2022-1202.notes_on_trinity.1.docx`](./notebook/AG.2022-1202.notes_on_trinity.1.docx)
[File: `AG.2022-1202.notes_on_trinity.2.bedtools_comparison.pdf`](./notebook/AG.2022-1202.notes_on_trinity.2.bedtools_comparison.pdf)

<a id="2-me-%E2%86%92-alison-friday-2022-1202-1726"></a>
#### 2: Me → Alison `Friday, 2022-1202, 17:26`
Thank you, Alison! I will check it soon. Have a great weekend!

-Kris

<a id="3-alison-%E2%86%92-me-friday-2022-1202-1749"></a>
#### 3: Alison → Me `Friday, 2022-1202, 17:49`
Awesome! I also went ahead and tried the right vs wrong analysis of the original annotation. Those notes are attached to this email. I saved the IGV regions for this in the shared folder within my Tsukiyama lab folder, so you can open these up for yourself if you want. All counting was rough and quick so numerical values might be inexact. 

Alison 

<a id="files-2"></a>
##### Files
[File: `AG.2022-1202.notes_on_trinity.3.right_vs_wrong_analysis.docx`](./notebook/AG.2022-1202.notes_on_trinity.3.right_vs_wrong_analysis.docx)

<a id="4-me-%E2%86%92-alison-saturday-2022-1203-910"></a>
#### 4: Me → Alison `Saturday, 2022-1203, 9:10`
This is great, thank you! Sometime next week, let's get you set up with a GitHub account so that you can get quick access to my work and notes for the project. No rush on this though; just sent you the GitHub invite because it had been on my mind.

-Kris
</details>
<br />

<a id="email-blacklist-file"></a>
### Email: *Blacklist file*
2022-1205-1207
<details>
<summary><i>Click to view</i></summary>

<a id="1-alison-%E2%86%92-me-monday-2022-1205-345-pm"></a>
#### 1: Alison → me `Monday, 2022-1205, 3:45 PM`
Hi Kris - 

You can find the .gff file for other genomics regions in `~/tsukiyamalab/alisong/annotation_files/other_regions` I downloaded the .fasta file from sgd, and then mapped it to the reference using gmap. The gff needs to be filtered to pick out the black list regions. We don't want to blacklist origins of replication or centromeres for example. Everything is labeled pretty clearly so I would just run a few quick awk scripts to filter, but I'm sure their are other ways too. Let know if you have any questions or issues. 

Here is the source for the fasta:  
[http://sgd-archive.yeastgenome.org/sequence/S288C_reference/other_features/](http://sgd-archive.yeastgenome.org/sequence/S288C_reference/other_features/)

Alison

<a id="2-me-%E2%86%92-alison-monday-2022-1205-405-pm"></a>
#### 2: Me → Alison `Monday, 2022-1205, 4:05 PM`
Thanks, this is great!

<a id="3-me-%E2%86%92-alison-tuesday-2022-1206-1110-am"></a>
#### 3: Me → Alison `Tuesday, 2022-1206, 11:10 AM`
Hi Alison,
 
It looks like there are 850 unique `Name=Something` features in the .gff file. Is there a quick way for me to know which to include in the blacklist and which to not? No rush at all. You can see the unique elements in `~/tsukiyamalab/alisong/annotation_files/other_regions/KA.other_features_genomic.sort-uniq-tally.txt`.
 
Also, do you think that blacklisting the telomeric repeats in this file will be enough to prevent the weird Trinity annotations we see in telomeric regions, or should we go with a complete 10-kb blacklisting of the starts and ends of chromosomes? Or perhaps both?
 
Thanks,  
Kris

<a id="4-me-%E2%86%92-alison-tuesday-2022-1206-1116-am"></a>
#### 4: Me → Alison `Tuesday, 2022-1206, 11:16 AM`
Ah—-I temporarily forgot about Christine’s telomere annotations. I think they’re from the UCSC table browser. If they look reasonable, perhaps we can just use those in place of blanket 10-kb blacklistings...

<a id="5-alison-%E2%86%92-me-tuesday-2022-1206-1144-am"></a>
#### 5: Alison → me `Tuesday, 2022-1206, 11:44 AM`
Hi Kris - 

Here are some notes on those region names. Let me know if you have any questions. 

Alison

<a id="note-fralison-on-what-to-blacklist-and-what-to-not"></a>
##### *Note fr/Alison on what to blacklist and what to not*
```txt
Region names. How I think we should deal with each category of region as based on how the name starts: 

ARS - Autonomously Replicating Sequence, aka origin of replication, don’t blacklist

CEN - centromere, don’t blacklist

HML/HMR - silent mating type cassette array, don’t blacklist

MATALPHA - mating type locus. We use mata instead of mat@ so this can be blacklisted. Maybe ask Toshi if it is worth adjusting the reference to the correct mating type. He will probably say no.

NTS - Non-transcribed region of the rDNA repeat, probably depleted by depletion probes but can blacklist

ORI - mitochondrial origin of replication, don’t need to blacklist

RE301 - recombination enhancer, don’t blacklist

TEL - telomere, use Christine’s

VDE - intein encoding region, don’t blacklist

Y - every yeast chromosome named as Y (yeast) A-P (chromosome 1-16, with 1 being A, and 16 being P) and L or R (left or right of centromere). Fairly sure these are all long terminal repeats of some kind and should be blacklisted.
```

<a id="6-me-%E2%86%92-alison-tuesday-2022-1206-1147-am"></a>
#### 6: Me → Alison `Tuesday, 2022-1206, 11:47 AM`

Very clear, thanks!

<a id="7-me-%E2%86%92-alison-tuesday-2022-1206-334-pm"></a>
#### 7: Me → Alison `Tuesday, 2022-1206, 3:34 PM`

Hi Alison,
 
This is just a rough-draft thought we can talk about when you’re back in the lab: I feel a little frightened or wary to blacklist all the LTRs since they and other REs (repetitive elements) make up parts of some genes, especially lncRNA-coding ones, and some other features like TFBS. (At least, that’s the case for mammals; I’m not up on the bio of REs in yeast, so correct me if this is incorrect.) Although I don’t know, I’m thinking that it could affect the transcripts that we build at those particular genes and features with Trinity.
 
Also, forgive my ignorance, but is Ty1 a specific RE or is it a class of REs containing a bunch of individually named REs?
 
Thanks,  
Kris

<a id="8-alison-%E2%86%92-me-wednesday-2022-1207-1220-pm"></a>
#### 8: Alison → me `Wednesday, 2022-1207, 12:20 PM`

Thinking about this more, I think it's reasonable to make the least conservative blacklist, use that and then only go more aggressive if there are still issues a more extensive blacklist could fix. 

Alison

<a id="9-me-%E2%86%92-alison-wednesday-2022-1207-1231-pm"></a>
#### 9: Me → Alison `Wednesday, 2022-1207, 12:31 PM`

By most conservative, do you mean all telomere regions in the file from Christine plus the stuff associated with “Y” as mentioned in the .rtf file you sent? (And possibly mat@ too?)

```txt
Y - every yeast chromosome named as Y (yeast) A-P (chromosome 1-16, with 1 being A, and 16 being P) and L or R (left or right of centromere). Fairly sure these are all long terminal repeats of some kind and should be blacklisted.
```

<a id="10-me-%E2%86%92-alison-wednesday-2022-1207-1239-pm"></a>
#### 10: Me → Alison `Wednesday, 2022-1207, 12:39 PM`

For a least conservative version, I think we’d be good to start with just the telomere regions in Christine’s file (also, we need to confirm the source of this file and how it was generated--my initial work suggests it’s from the UCSC table browser, but I’m not yet certain about that).
 
For a more conservative version, I think we’d be good to try the regions in Christine’s file, Ty1-* sequences, and telomeric-repeat sequences.
 
For a very conservative version, I think we’d be good to try the above plus all other LTRs (or whatever each “Y” feature is).
 
In IGV, do you see any out-of-place spikes in 4tU-seq signal that look like they could be the result of non-specific alignment?
</details>
<br />

<a id="email-bam-files-in-directory-~tsukiyamalabalisongkris_bams"></a>
### Email: *Bam files in directory `~/tsukiyamalab/alisong/Kris_bams/`*
2022-1212
<details>
<summary><i>Click to view</i></summary>

<a id="1-me-%E2%86%92-alison-monday-2022-1212-327-pm"></a>
#### 1: Me → Alison `Monday, 2022-1212, 3:27 PM`
Hi Alison,

I created a subdirectory in your directory called 'Kris_bams/' ('~/tsukiyamalab/alisong/Kris_bams/'). Inside 'Kris_bams/', you can find symlinks to all of the relevant (and less or not relevant) bams.

In 'Kris_bams/', the bams are subdivided into three subdirectories:
- unprocessed/
- preprocessed/
- preprocessed-full/

On "unprocessed" bams...
- reads adapter- and quality-trimmed by trim_galore: FALSE
- reads k-mer-corrected by rcorrector: FALSE
- STAR alignment type: "Local" (allows "soft clipping")

On "preprocessed" bams...
- reads adapter- and quality-trimmed by trim_galore: TRUE
- reads k-mer-corrected by rcorrector: FALSE
- STAR alignment type: "EndToEnd" (doesn't allow "soft clipping")

On "preprocessed (full)" bams...
- reads adapter- and quality-trimmed by trim_galore: TRUE
- reads k-mer-corrected by rcorrector: TRUE
- STAR alignment type: "EndToEnd" (doesn't allow "soft clipping")

Within each of these subdirectories, you'll find bams for...
- 5781_G1_IN
- 5781_G1_IP
- 5781_Q_IN
- 5781_Q_IP
- 5782_G1_IN
- 5782_G1_IP
- 5782_Q_IN
- 5782_Q_IP

There are four of each kind
- multi-hit-mode_1000: Up to 1000 alignments are allowed/retained for a given read
- multi-hit-mode_100: Up to 100 alignments are allowed/retained for a given read
- multi-hit-mode_10: Up to 10 alignments are allowed/retained for a given read
- multi-hit-mode_1: Only 1 alignment are allowed/retained for a given read (standard; basically, "multi-hit mode" was off)

I'd start with making bigwig, etc. files for "unprocessed", "preprocessed", and "preprocessed (full)" 5781_Q_IP multi-hit-mode_1 bams (three samples). From there, you could branch out to 5782_Q_IP multi-hit-mode_1 bams (another three samples). Then maybe the same for Q_IN (another six samples)?

From there, maybe try 5781_Q_{IP,IN} multi-hit-mode_1000 and 5782_Q_{IP,IN} multi-hit-mode_1000 bams (twelve samples) to see where all the multimappers go (e.g., are they going to Ty elements?)

Thanks,  
Kris

P.S. Symbolic links stuff:
 
General pattern: `ln -s source_file target_file`
 
For example, here’s how I quickly symlinked the files to, e.g., 'Kris_bams/preprocessed-full/'
```bash
#!/bin/bash
#DONTRUN
 
pwd
# /home/kalavatt/tsukiyamalab/alisong/Kris_bams/preprocessed-full
 
p_pf="/home/kalavatt/tsukiyamalab/kalavatt/2022_transcriptome-construction/results/2022-1201/files_processed-full/bam_trim-rcor-cor_split/EndToEnd"
for i in "${p_pf}/"*".sc_all.bam"; do
       echo "Working with ${i}"
       # echo "${i}" "$(basename "${i}")"  # Test things before actually running the command
       ln -s "${i}" "$(basename "${i}")"  # Run the command when things look reasonable
       echo ""
done
```

<a id="2-alison-%E2%86%92-me-monday-2022-1212-342-pm"></a>
#### 2: Alison → Me `Monday, 2022-1212, 3:42 PM`
Awesome! Thank you so much!

Alison
</details>
<br />

<a id="email-gff3-files-in-~tsukiyamalabalisongkrisgtfs_2022-1214param_gene-overlap_falsetrue"></a>
### Email: *Gff3 files in `~/tsukiyamalab/alisong/Kris/gtfs_2022-1214/param_gene-overlap_{FALSE,TRUE}`*
2022-1213
<details>
<summary><i>Click to view</i></summary>

<a id="1-me-%E2%86%92-alison-wednesday-2022-1213-317-pm"></a>
#### 1: Me → Alison `Wednesday, 2022-1213, 3:17 PM`
Hi Alison,
 
I created symlinks to all of the relevant (and less or not relevant) gff3s in ~/tsukiyamalab/alisong/Kris/gtfs_2022-1214/.
 
Within this directory, there are two subdirectories:
param_gene-overlap_FALSE/
param_gene-overlap_TRUE/
 
Documentation is forthcoming (for the time being, we can start with files in param_gene-overlap_FALSE/, although comparing to files in *_TRUE/ will be important).
 
There are nine kinds of files inside each subdirectory. Here’s the breakdown (GF = genome-free mode, GG = genome-guided mode):
__"unprocessed"__
+ PASA from Trinity GF + Trinity GG multi-hit-mode 1
+ PASA from Trinity GF + Trinity GG multi-hit-mode 10
+ PASA from Trinity GF + Trinity GG multi-hit-mode 100
__"processed"__
+ PASA from Trinity GF + Trinity GG multi-hit-mode 1
+ PASA from Trinity GF + Trinity GG multi-hit-mode 10
+ PASA from Trinity GF + Trinity GG multi-hit-mode 100
__"processed (full)"__
+ PASA from Trinity GF + Trinity GG multi-hit-mode 1
+ PASA from Trinity GF + Trinity GG multi-hit-mode 10
+ PASA from Trinity GF + Trinity GG multi-hit-mode 100
 
Trinity GF is always done with multi-hit-mode 1 (because GF doesn’t know how to handle multi-mappers, although Trinity GG does). I’ll be sure to send you rationale, experiment details, etc. before you come into the office tomorrow.
 
Thanks,
Kris
</details>
<br />

<a id="email-preprocessing-thoughts"></a>
### Email: *Preprocessing Thoughts*
2022-1215-1216
<details>
<summary><i>Click to view</i></summary>

<a id="1-alison-%E2%86%92-me-thursday-2022-1215-1907"></a>
#### 1: Alison → Me `Thursday, 2022-1215, 19:07`
I pulled up and looked at some of the bigwigs you made and I have pretty strong opinions almost immediately on what preprocessing is going to benefit Trinity. I made a powerpoint about it for clarity sake.

Far and away the most important variable seems to be how many hits are allowed. I only visually compared 1 and 10, but I will plan to look at 100 and 1000 tomorrow. That said, I think we want to retain multiple hits, do no MapQ filtering, and then implement our blacklist for regions of concern between alignment and Trinity. Hopefully the attached presentation can illustrate why I have come to that initial conclusion.

I submitted my library tonight so I am now full time computational for the next few weeks minimum. I am planning to check out the gffs tomorrow to see if trinity is affected in a straightforward way by the various modes of preprocessing.

Let me know if you have any questions/issues or else if you have a different idea about what will work best.

Alison

<a id="files-3"></a>
##### Files
[File: `AG.2022-1215.preprocessing-thoughts.pptx`](./notebook/AG.2022-1215.preprocessing_thoughts.pptx)

<a id="2-me-%E2%86%92-alison-friday-2022-1216-1207"></a>
#### 2: Me → Alison `Friday, 2022-1216, 12:07`
Wow, I’m happy to see that the retention of multi-mappers seems to better represent signal at paralogs and other features, and I agree with all your comments. I like your thinking re: keeping as many multi-mappers as is reasonable and addressing false signal at repetitive regions through blacklisting rather than filtering for multi-mappers and/or MAPQ.

I symlinked a ton of .bams and made a ton of .bws today too; now, I’m working on automating the creation of .xml files for IGV so that we don’t have to do so much drag and dropping and track set-up work.

-Kris
</details>
<br />

<a id="email-split-script"></a>
### Email: *split script*
2022-1216
<details>
<summary><i>Click to view</i></summary>

<a id="1-alison-%E2%86%92-me-friday-2022-1216-1450"></a>
#### 1: Alison → Me `Friday, 2022-1216, 14:50`
Hi Kris! 

This splits the strands of .gff3 files, but could easily be re-written for bed files or gtfs as well. It ran on 9 gff files in about 1 second, so I would not spend time optimizing this for speed, as it is already quite fast. 

Also it does look like I set up the bamcoverage tool from deeptools to split the stands wrong (I think you mentioned in a previous email, and I missed it). This can probably be fixed just by renaming filter strand forward to "minus", and reverse to "plus". We could also go back to the samtools way we used to do it. Not a huge issue today, but also not something we want to forget about. 

Alison

<a id="files-4"></a>
##### Files
[File: `AG.2022-1216.awk_split_strand.sh`](./notebook/AG.2022-1216.awk_split_strand.sh)
</details>
<br />

<a id="email-more-trinity-thoughts"></a>
### Email: *More trinity thoughts*
2022-1216-1219
<details>
<summary><i>Click to view</i></summary>

<a id="1-alison-%E2%86%92-me-friday-2022-1216-1813"></a>
#### 1: Alison → Me `Friday, 2022-1216, 18:13`
I implemented the bedtools strategy we discussed and my intuition that rcor was more helpful than hurtful seems to be correct. Powerpoint attached with more details. There is really absolutely no reason to work on this over the weekend.

Alison

<a id="files-5"></a>
##### Files
[File: `AG.2022-1216.trinity_behavior.pptx`](./notebook/AG.2022-1216.trinity_behavior.pptx)

<a id="2-me-%E2%86%92-alison-monday-2022-1219-1149"></a>
#### 2: Me → Alison `Monday, 2022-1219, 11:49`
Thanks, Alison. This is great info. Apologies for the very long message that follows: I took time to consolidate my thoughts and finish writing an overview of important parameters for us to consider moving forward; that, in turn, required me to finish writing a brief overview of what `PASA` does, which is also important for informing our next steps. (Both those things were on my to-do list, so I can now scratch them off.)

So, everything in your *‘Things that might help’* (slide 4 of `trinity_behavior.pptx`) bullet is reasonable and either easy or relatively easy to implement. (However, I need to study more on the UMI filtering. If I remember correctly, I was trying to get straightforward answers to questions such as, ‘What are the required infiles, which `umitools` functions do I run, do I need to explicitly specify the UMIs, etc.?’).

After writing up this message, I will begin implementing the rcor-only processing (i.e., preprocessing of reads without running `trim_galore`) and `STAR` multi-hit-modes for ‘5’ and ‘15’ (and perhaps other values).

Another thing: Sorry, I’m not understanding your `bedtools` experiments: I must be missing something. When you’re in the office, do you mind explaining them to me again?


\#  ===========================================================  
Anyway, if we think of the general, overarching process for this transcriptome-assembly work in three steps, where <u>step 1</u> is generating/preprocessing the infiles for running `Trinity` GF and GG modes, <u>step 2</u> is running `Trinity` GG and GF modes, and <u>step 3</u> is running `PASA` with the `Trinity` GG and GF outfiles, then I think we should prioritize optimizing <u>steps 1 and 3</u> prior to tuning experiments with <u>step 2</u>. In Brian’s papers, documentations, and comments in the community message boards, the default `Trinity` parameters are shown to do “well enough” under most conditions, including those for both gene-sparse and gene-dense genomes (provided we use stranded RNA-seq and invoke `--jaccard_clip`). So, in addition to that, I go over my reasoning for that assessment below and, based on your work over the last few days, I think we’re not far from achieving the goal of optimizing <u>steps 1 and 3</u>. TLDR: I think that the correct tuning of `PASA` parameters ‘`--stringent_alignment_overlap`’ and ‘`--gene_overlap`’ has strong potential to achieve the goals we want for the transcriptome assembly. Another reason to get <u>steps 1 and 3</u> reasonably “right” is that I think the parameters will hold steady regardless of the source/conditions of the infile, i.e., they won’t need to be changed if we’re using Q, G1, IP, IN, one knockout model, another knockout model, etc. I am not so sure about that for <u>step 2</u> and would want to message Brian about it.

So, we’re now on our way to determining ideal conditions for <u>step 1</u>. But regarding <u>step 3</u>, the running of `PASA`, there are two parameters to check and consider adjusting before diving into any tuning of `Trinity` parameters. One parameter is ‘`--gene_overlap`’ and the other is ‘`--stringent_alignment_overlap`’. Here’s what they do, which informs why I think they deserve our attention:

- `--stringent_alignment_overlap`
	+ “`(suggested: 30.0) overlapping transcripts must have this min % overlap to be clustered`” (from `Launch_PASA_pipeline.pl --help`)
	+ By default, `PASA` clusters alignments\* with minimal overlap (e.g., if there’s a situation in which two transcripts have only a tiny amount of overlap, they are nonetheless clustered together and become one gene/feature ~~annotation~~transcript in the assembly), but this may not be ideal for gene-dense genomes such as *S. cerevisiae*
	+ Since *S. cerevisiae* gene density is quite high, and since we expect transcripts from neighboring genes to overlap often, especially in their UTRs, I ran `Launch_PASA_pipeline.pl` with ‘`--stringent_alignment_overlap 30.0`’ (the value suggested by Brian in both the documentation and in my messages with him)
	+ However, based on what you’re seeing, it is perhaps worth testing adjustments to this value or removing this parameter altogether (and thus running `PASA` in its default mode)
		* My thinking here—and please definitely correct me if I’m wrong—is that, if we see expression that covers multiple genes/features, we want big, long `Trinity`/`PASA`-determined transcripts that reflect those and not necessarily the individual genes (I guess these kinds of `Trinity`/`PASA`-determined transcripts would appear unnecessarily fragmented to us, right?)
		* This is what, if I remember correctly when I was looking at things in `IGV` with you at your computer, we want, right?
	+ Let’s talk about the above and, in general, `--stringent_alignment_overlap` when we’re both in the office
- `--gene_overlap`
	+ From `Launch_PASA_pipeline.pl --help`: “`(suggested: 50.0) transcripts overlapping existing gene annotations [e.g., "Saccharomyces_cerevisiae.R64-1-1.108.gff3"] are clustered; intergenic alignments are clustered by default mechanism`”
	+ With the `param_gene-overlap_TRUE/` outfiles, I ran `Launch_PASA_pipeline.pl` with ‘`--stringent_alignment_overlap 30.0 --gene_overlap 50.0`’
	+ With `param_gene-overlap_FALSE/` outfiles, I ran `Launch_PASA_pipeline.pl` with ‘`--stringent_alignment_overlap 30.0`’ and no call to ‘`--gene_overlap`’
		* So, with these outfiles, `PASA` was run in such a way that it did not reference any provided annotations (in our case, "`Saccharomyces_cerevisiae.R64-1-1.108.gff3`")
		* And thus, alignments in gene-poor regions were clustered with the default mechanism of “any-percent overlap,” and alignments in gene-dense regions were clustered such that they must have a minimum overlap of 30% to be clustered (i.e., considered one gene/feature ~~annotation~~transcript and not separate gene/feature ~~annotation~~transcript)

I think it will be good if you do rough comparisons of selected data from `param_gene-overlap_TRUE` (what I think you’ve looked at already) versus `param_gene-overlap_FALSE`. For example, are we roughly seeing more of what we want with one versus the other?


\#  --------------------------  
\*A rough overview of what PASA does:  
1. *Optional step:* Identify evidence of polyadenylation in Trinity GF and GG .fasta infiles
	a. If evidence of polyadenylation is found, then strip the poly-A and discard low-quality sequences
	b. *I ran this optional step*
2. Simultaneously align Trinity GF and GG .fasta files to a reference genome using gmap and/or blat and/or minimap2
3. Identify and separate valid alignments from invalid alignments
	a. PASA uses only “near-perfect” alignments, i.e., those that align with a percent identity of ≥95% along a high percentage, ≥90%, of the transcript length
	b. *The following is not necessarily relevant for our work but is relevant to understand what PASA is doing:* Each alignment is required to have consensus splice sites at all inferred intron boundaries
4. Assemble the alignments
	a. Valid transcript alignments are clustered based on genome mapping location and assembled into gene/feature structures
	b. *The following is not necessarily relevant for our work but is relevant to understand what PASA is doing:*
    	i. Where there is evidence of alternative transcript from, e.g., splicing events, transcript alignments are assembled into gene structures that include the maximal number of compatible transcript alignments
    	ii. Here, compatible transcript alignments are defined as those with identical gene/feature structures in their regions of overlap
5. Group alternatively spliced isoforms (again, *not necessarily relevant for our work but relevant to understand what PASA is doing*)
6. Alignment assemblies that (i) map to the same genomic locus, (ii) significantly overlap, and (iii) are transcribed on the same strand are grouped into clusters of assemblies
7. Automatic genome annotation
	a. *I haven’t done this or really investigated how PASA does this yet*
	b. `#TODO` `#MAYBE`


\#  ===========================================================  
Once we’re fairly secure in our choices for <u>steps 1 and 3</u>, we can, if necessary, move on to running experiments to tune `Trinity` parameters such as `--min_glue`. As a reminder, here’s the documentation snippet for `--min_glue`:
>"`--min_glue` will increase the stringency of the `inchworm` contig clustering, and ... can result in more fragmented assemblies. Decreasing it can cause other problems, such as over-clustering."

In my notes, I wrote to myself to potentially test values of ‘1’, ‘2’, and ‘4’ for `--min_glue`; this is based on the parameter optimization work of [McIlwain et al.](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4889671/) So far, we’ve looked at results with ‘1’ (the default setting for `--min_glue`); soon, we can perhaps move on to ‘2’ and ‘4’. `--min_glue` is like PASA’s `--stringent_alignment_overlap` and `--gene_overlap` in that increasing their values can also result in “more fragmented” assemblies.


\#  --------------------------  
In addition to `--min_glue`, `Trinity` parameters to consider testing, alone or in combination, are `--min_kmer_cov`, `--min_iso_ratio`, `--glue_factor`, and `--normalize_max_read_cov`:

- `--min_kmer_cov`
	+ "...with a setting of 2, it means that singleton k-mers will not be included in initial `Inchworm` contigs (suggested by the `Trinity` team)"
	+ Values to potentially test: 1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024
	+ Default: 1
- `--min_iso_ratio`
	+ "`--min_iso_ratio` just takes the kmer abundance into account when doing the `inchworm` contig clustering, mainly to prevent very lowly expressed contigs from being clustered together with highly expressed contigs"
	+ Values to potentially test: 0.01, 0.05, #TBD
	+ Default: 0.05
- `--min_glue`
	+ "`--min_glue` will increase the stringency of the `inchworm` contig clustering, and ... can result in more fragmented assemblies. Decreasing it can cause other problems, such as over-clustering."
	+ Values to potentially test: 1, 2, 4
	+ Default: 2
- `--glue_factor`
	+ "`--glue_factor` increases the `--min_glue` value on a per inchworm contig pair basis such that it increases the support required based on the kmer coverage of each of the contigs. That is to reflect that more highly expressed contigs (higher kmer coverage) should have more evidence (reads) when linking them together."
	+ Values to potentially test: 0.01, 0.05, #TBD
	+ Default: 0.05
- `--normalize_max_read_cov`
	+ "It means that 'poorly covered regions are unchanged, but reads are down-sampled in high-coverage regions (see slide 16 at link)"
	+ Values to potentially test: 25, 50, 100, 200
	+ Default: 200


\#  ===========================================================  
Also, I didn’t work on anything over the weekend `:D`.

-Kris
</details>
<br />

<a id="email-update"></a>
### Email: *Update*
2022-1221-1222
<details>
<summary><i>Click to view</i></summary>

<a id="1-me-%E2%86%92-alison-monday-2022-1221-1202"></a>
#### 1: Me → Alison `Monday, 2022-1221, 12:02`
Hi Alison,

Thanks for the great info on the alignments—I will generate files using local (soft-clipping) alignment as you suggested. I think it has potential to improve our downstream transcript constructions.

Yesterday, I figured out the issue for why our combined --gene_overlap/--stringent_alignment_overlap files looked the same as our --stringent_alignment_overlap files: It turns out that the arguments --gene_overlap and --stringent_alignment_overlap are mutually exclusive, and if both are called at the same time, then --stringent_alignment_overlap takes precedence. This was not really discussed in or clear from the PASA documentation; I touched base with Brian, and he will update the documentation to make this clear. 

Now, I'm setting up and running experiments for --gene_overlapalone, --stringent_alignment_overlap, and default (i.e., neither --gene_overlap nor --stringent_alignment_overlap).

Also, just want to give a heads up that our symlinks in gff3s_2022-1214/param_gene-overlap_TRUE are now (or soon will be) broken, but the ones in gff3s_2022-1214/param_gene-overlap_FALSE should still be working.

Going to try to get you some more gff3s for analysis before I'm out and about tonight and tomorrow.

Thanks,
Kris

<a id="2-alison-%E2%86%92-me-monday-2022-1221-1207"></a>
#### 2: Alison → Me `Monday, 2022-1221, 12:07`
Awesome! Enjoy your trip and holiday! If you aren't able to get me new files, it's totally okay. 

Alison 

<a id="3-me-%E2%86%92-alison-monday-2022-1221-1750"></a>
#### 3: Me → Alison `Monday, 2022-1221, 17:50`
Thank you! Each PASA job is taking about 4–6 hours to complete with 8 CPUs per job, and most are still running or are in queue status right now. I expect that the jobs will run through the night and most, if not all, of the gff3s will be ready tomorrow. So, I should have most or all of them organized and symlinked for you by the time you come into lab.

From there, I'll move on to writing/adjusting code for the other suggestions you mentioned (e.g., multi-hit modes with 5 and 15 multi-mappers, rcor with soft-clipping allowed, etc.) and the blacklist work. Then, I'll begin to dive into some of the eyeballing work that you're doing now, and I'll use your bedtools scripts to run the same calculations that you are. Between that and the need to put together a PowerPoint presentation, I suspect I'll have plenty to do during your week off.

Thanks again,
Kris

<a id="4-me-%E2%86%92-alison-tuesday-2022-1222-1157"></a>
#### 4: Me → Alison `Tuesday, 2022-1222, 11:57`
Hi Alison,

Many of the PASA jobs are taking a very long time to complete (≥22 hours). This is surprising to me. I am looking into it now; so far, everything seems OK, just slow.

Anyway, I'm sending you the gff3s from the jobs that have already completed. They're in a zip file; I didn't want to do the symlinking until all jobs have completed.

Here's how to delineate experiment conditions:
- PASA parameters
	+ stringent-alignment-overlap = PASA run with --stringent_alignment_overlap *(many of the jobs have not yet been completed)*
	+ gene-overlap = PASA run with --gene_overlap *(most of the jobs have not yet completed)*
	+ minimal-overlap = PASA run with neither of the above, i.e., the default method *(these were the last-submitted jobs, so none have completed yet)*
- read preprocessing
	+ .un_ = unprocessed data
	+ .trim.un_ = trim_galore-processed data
	+ .trim-rcor. = trim_galore- and rcor-processed data
	+ *haven't yet generated the other datasets we talked about*

For each un- or preprocessed sample, I ran PASA --stringent_alignment_overlap with 10.0 20.0 30.0 40.0 50.0 60.0 70.0 80.0 90.0. I did the same for PASA --gene_overlap. I was thinking we can start with simple comparisons and work from there, starting from, for example, 30.0 for PASA --stringent_alignment_overlap and 50.0 for PASA --gene_overlap, and then branching out as needed. No need to examine everything, I think: just as needed.

It may be useful to quickly eyeball some the datasets that are available now, despite being incomplete, and despite some perhaps being not the most useful values for --stringent_alignment_overlap and --gene_overlap.

Thanks,
Kris

<a id="5-me-%E2%86%92-alison-tuesday-2022-1222-1200"></a>
#### 5: Me → Alison `Tuesday, 2022-1222, 12:00`
Also, if you have any questions or anything, cc kalavattam@gmail.com: I'll see them sooner.

<a id="6-alison-%E2%86%92-me-tuesday-2022-1221-1244"></a>
#### 6: Alison → Me `Tuesday, 2022-1221, 12:44`
Hi Kris -

I think the server is just slow right now since so many people are remote between the holidays and snowy Seattle. I will try and take a look at the files sometime this afternoon. For future reference, could you add the relevant settings to the files themselves, not just the folder names? It makes it easier for me to know what I am looking at in IGV. 

Feel free to email me while I am out -- my work email goes to my phone, and I will monitor it in case anything comes up. I am trying not to do a ton of computation work over the holiday, but if I am tempted and learn anything interesting, I will let you know. 

Alison

<a id="7-alison-%E2%86%92-me-tuesday-2022-1222-1412"></a>
#### 7: Alison → Me `Tuesday, 2022-1222, 14:12`
I went ahead and tried to compare different PASA settings with the same alignment conditions. I was able to use multi hit mode 10, local alignment and compare stringent 60, stringent 80 and gene overlap 50 - of these gene overlap 50 seemed to do better by a wide margin using my mRNA test (1470 correct vs 1316 correct = 11% improvement). I would be curious if this pattern holds across alignment methods/PASA settings, but it seems like those jobs haven't fully run yet so it might be hard to tell. 

Alison
</details>
<br />

<a id="email-preprocessing---more-thoughts"></a>
### Email: *Preprocessing - more thoughts!*
2022-1222
<details>
<summary><i>Click to view</i></summary>

<a id="1-alison-%E2%86%92-me-tuesday-2022-1222-1655"></a>
#### 1: Alison → Me `Tuesday, 2022-1222, 16:55`
I went ahead and looked at the bams directly in IGV, and it was pretty interesting. I think there might be some additional alignment settings that matter a lot (although if they matter to trinity is another matter). Let me know what you think when you get the chance.

Enjoy your trip and no rush on this! I may or may not email you more before I am off - will depend on if I find anything else noteworthy.

Best
Alison

<a id="files-6"></a>
##### Files
[File: `AG.2022-1222.preprocessing_more_thoughts.pptx`](./notebook/AG.2022-1222.preprocessing_more_thoughts.pptx)
</details>
<br />
